---
title: "05-metabolomics-analysis"
author: "Yaamini Venkataraman"
date: '2023-12-18'
output: html_document
---

In this script, I will analyze metabolomics data from the West Coast Metabolomics Center (UC Davis). I will conduct an initial assessment of the data (PLS-DA), understand important metabolites and pathways (VIP analysis and WGCNA), and prepare for additional network analysis through MetaboAnalyst. This workflow and script is modified from [Huffmyer et al. (2023)](https://www.biorxiv.org/content/10.1101/2023.03.20.533475v2.full) and [Trigg et al. (2019)](http://dx.doi.org/10.1038/s41598-019-46947-6).

# Set up R Markdown document

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = normalizePath("../output/05-metabolomics-analysis/")) #Set root directory
```

```{r}
getwd()
```

# Install packages

```{r packages, warning = FALSE, message = FALSE}
# if ("tidyverse" %in% rownames(installed.packages()) == 'FALSE') install.packages('tidyverse')
# if ("vegan" %in% rownames(installed.packages()) == 'FALSE') install.packages('vegan')
# if (!requireNamespace("BiocManager", quietly = TRUE))
#   install.packages("BiocManager")
# if ("mixOmics" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("mixOmics")
# if ("RVAideMemoire" %in% rownames(installed.packages()) == 'FALSE') install.packages('RVAideMemoire')
# if ("broom" %in% rownames(installed.packages()) == 'FALSE') install.packages('broom')
# if ("RColorBrewer" %in% rownames(installed.packages()) == 'FALSE') install.packages('RColorBrewer')
# if ("pheatmap" %in% rownames(installed.packages()) == 'FALSE') install.packages('pheatmap')
# if ("genefilter" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("genefilter")
# if ("WGCNA" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('WGCNA')
# if ("dendsort" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('dendsort')
# if ("ComplexHeatmap" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('ComplexHeatmap')
# if ("svglite" %in% rownames(installed.packages()) == 'FALSE') install.packages("svglite")
# if ("ggfortify" %in% rownames(installed.packages()) == 'FALSE') install.packages("ggfortify")
# if (!requireNamespace("devtools", quietly = TRUE))
#     install.packages("devtools")
# devtools::install_github("andjar/ALASCA", ref = "main")
require(tidyverse)
require(vegan)
require(mixOmics)
require(RVAideMemoire)
require(broom)
require(RColorBrewer)
require(pheatmap)
require(genefilter)
require(WGCNA)
require(dendsort)
require(ComplexHeatmap)
require(svglite)
require(ggfortify)
require(devtools)
require(ALASCA)
```

```{r}
source("../../code/biostats.R") #Source multivariate analysis functions
```

```{r}
sessionInfo()
```

# Import data

```{r}
rawMetabolomicsData <- read.csv("../../data/raw_metabolomics_data.csv", header = TRUE) %>% 
  dplyr::rename_with(~stringr::str_replace_all(., "X", "")) #Import data. Remove the "X" prefix for crab IDs
head(rawMetabolomicsData) #Confirm import
```

```{r}
metabolomicsMetadata <- read.csv("../../data/metabolomics_metadata.csv", header = FALSE, strip.white = TRUE) %>% 
  dplyr::rename_with(~stringr::str_replace_all(., "V", "")) %>%
  t(.) %>% as.data.frame(.) %>% 
  purrr::set_names(as.character(dplyr::slice(., 1))) %>%
  dplyr::slice(., -1) %>% 
  mutate(., unite(., "treatment_day", treatment:day, sep = "_")) %>%
  mutate(., treatment_day = gsub(x = treatment_day, pattern = "_  3", replacement = "_3")) %>%
  mutate(., treatment_day = gsub(x = treatment_day, pattern = "_ 22", replacement = "_22")) %>%
  mutate(., treatment_day = gsub(x = treatment_day, pattern = " 13_", replacement = "13_")) %>%
  mutate(., treatment_day = gsub(x = treatment_day, pattern = " 30_", replacement = "30_")) %>%
  mutate(., treatment_day = gsub(x = treatment_day, pattern = "  5_", replacement = "5_")) #Import metadata. Do not include any header information. Remove "V" from column names. Transform so crab.ID, treatment, and day are all columns. Set the first row in the dataframe as column names, then remove that row.
tail(metabolomicsMetadata) #Confirm import
```

# Format data

Before analyzing the data, I will transform the data and match with treatment metadata. The West Coast Metabolomics Center has already normalized data using the average mTIC (sum of all peak heights for *identified* metabolites) of the samples. 

```{r}
transMetabData <- rawMetabolomicsData %>%
  dplyr::select(., -c(2:8)) %>%
  column_to_rownames(., var = "BinBase.name") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(., var = "crab.ID") %>% 
  left_join(metabolomicsMetadata, ., by = "crab.ID") #Remove columns with MS information, convert BinBase.name to rownames, and transform dataframe. Convert rownames to crab.ID column
head(transMetabData) #Confirm changes
```

# Initial data investigation with PCA

```{bash}
mkdir PCA
mkdir PCA/figures
```

```{r}
scaled.pca <- prcomp(transMetabData[c(5:601)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca) #Standard deviations for each PC
```

```{r PCA interpretation}
pca.eigenval(scaled.pca) #Eigenvalues greater than the broken-stick expectation are considered to explain a statistically significance proportion of the variance in the original dataset. PC3 and PC4 eigenvalue > BSV
ordiMonteRandTest <- ordi.monte(transMetabData[c(5:601)], ord = "pca", dim = 6, plot = FALSE) #Conduct monte carlo randomization test using 1000 random permutations of the input data.
ordiMonteRandTest

#Will move forward with PC1-2
```

## Plot PCA

```{r}
plotColors <- c(brewer.pal(9, "Reds")[7],
                brewer.pal(9, "Greys")[7],
                brewer.pal(9, "Blues")[7]) #Create color scheme
```

### All metabolites

```{r}
pcaPlot1 <- autoplot(scaled.pca, data = transMetabData, x = 1, y = 2,
                     size = 3, alpha = 0.8, colour = "treatment", shape = "treatment", 
                     frame = TRUE, frame.colour = "treatment", loadings = FALSE) +
  scale_color_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                     name = "Temperature (ºC)") +
  scale_fill_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                    name = "Temperature (ºC)") +
  scale_shape_manual(values = c(15, 19, 17),
                     name = "Temperature (ºC)") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each treatment Do not include loadings. Manually define color, fill, and shapes.
pcaPlot1
ggsave("PCA/figures/all-data-PCA-treatment.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot2 <- autoplot(scaled.pca, data = transMetabData, x = 1, y = 2,
                     size = 5, alpha = 0.8, colour = "day", shape = "day",
                     frame = TRUE, frame.colour = "day", loadings = FALSE) +
  scale_color_manual(values = c("black", "purple"),
                     name = "Day") +
  scale_fill_manual(values = c("black", "purple"),
                    name = "Day") +
  scale_shape_manual(values = c(20, 18),
                     name = "Day") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Same as above but color by day and not treatment
pcaPlot2
ggsave("PCA/figures/all-data-PCA-day.pdf", height = 8.5, width = 11)
```

```{r}
pcaPlot3 <- autoplot(scaled.pca, data = transMetabData, x = 1, y = 2,
                     size = 4, alpha = 0.8, colour = "treatment", shape = "day", 
                     frame = TRUE, frame.colour = "treatment", loadings = FALSE) +
  scale_color_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                     name = "Temperature (ºC)") +
  scale_fill_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                    name = "Temperature (ºC)") +
  scale_shape_manual(values = c(20, 18),
                     name = "Day") +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot3
ggsave("PCA/figures/all-data-PCA-day-treatment.pdf", height = 8.5, width = 11)
```

Overall, it seems like metabolite profiles are segregating by treatment, with the 30ºC profiles more segregated from the 5ºC and 13ºC profiles. It also seems like metabolite profiles have higher spread over time, potentially due to the impact of prolonged temperature exposure on physiology.

### Identified metabolites

I'm going to make a very similar PCA to the one above, but only using metabolites identified by the sequencer.

```{r}
transMetabData[c(5:156)] #All known metabolites
```


```{r}
scaled.pca.id <- prcomp(transMetabData[c(5:156)], scale = TRUE, center = TRUE) #Use prcomp to perform a centered and scaled PCA calculation
summary(scaled.pca.id) #Standard deviations for each PC
```

```{r}
pcaPlot4 <- autoplot(scaled.pca.id, data = transMetabData, x = 1, y = 2,
                     size = 4, alpha = 0.8, colour = "treatment", shape = "day", 
                     frame = TRUE, frame.colour = "treatment", loadings = FALSE) +
  scale_color_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                     name = "Temperature (ºC)") +
  scale_fill_manual(values = c(plotColors[3], plotColors[2], plotColors[1]),
                    name = "Temperature (ºC)") +
  scale_shape_manual(values = c(18, 20),
                     name = "Day",
                     labels = c("4", "22")) +
  theme_classic(base_size = 15) + theme(legend.position = "right",
                                        plot.background = element_blank()) #Use autoplot to plot the prcomp object. Color points and confidence ellipses by treatment, and have different shapes for each day. Do not include loadings. Manually define color, fill, and shapes.
pcaPlot4
ggsave("PCA/figures/known-metabolites-PCA-day-treatment.pdf", height = 8.5, width = 11)
```

## Conduct PERMANOVA

I'll use a global permutational multivariate analysis of variance (PERMANOVA) to determine if treatment and day had a significant influence on metabolite profiles. The significant effects from the PCA will be used in the downstream PLS-DA to interrogate trends.

### All metabolites

```{r}
dissim.metab <- vegdist(scale(transMetabData[c(5:601)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.metab <- adonis2(dissim.metab ~ as.factor(treatment)*as.numeric(day), data = transMetabData, method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles. Include "by = "terms"" to ensure test assess significance for each term.
permanova.metab #Significant influence of all factors
```

```{r}
broom::tidy(permanova.metab) %>%
  write.csv(., "PCA/all-data-PERMANOVA-results.csv", quote = FALSE) #Save PERMANOVA output
```

I will need to use a PERMDISP test to determine if significant differences are due to centroid or variance differences. I will do a separate test for each variable.

```{r}
disp.metab.temp <- betadisper(dissim.metab, group = as.factor(transMetabData$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.metab.temp) #Dispersion differences exist
boxplot(disp.metab.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.metab.temp) #Variance is different between groups. Temperature significance is due to centroid and dispersion differences!
```

```{r}
disp.metab.day <- betadisper(dissim.metab, group = as.factor(transMetabData$day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.metab.day) #Similar dispersion, but some clear outliers for day 22
boxplot(disp.metab.day) #Similar dispersion differences, outliers seen more easily
anova(disp.metab.day) #Variance is the same between groups. Day significance is due to centroiddifferences!
```

```{r}
disp.metab.tempDay <- betadisper(dissim.metab, group = as.factor(transMetabData$treatment_day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.metab.tempDay) #Dispersion differences exist, especially with 30_22
boxplot(disp.metab.tempDay) #Very clear dispersion differences for each unique treatment!
anova(disp.metab.tempDay) #Variance is different between groups. Interaction significance is due to centroid and dispersion differences!
```

```{r}
rbind(broom::tidy(anova(disp.metab.temp)) %>%
        mutate(var = "temperature"),
      broom::tidy(anova(disp.metab.day)) %>%
        mutate(var = "day"),
      broom::tidy(anova(disp.metab.tempDay)) %>%
        mutate(var = "interaction")) %>%
  write.csv("PCA/all-data-PERMDISP-results.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```

### Identified metabolites

I'm going to repeat the same analysis, but only look at trends for metabolites identified by the West Coast Metabolomics Center.


```{r}
dissim.metabid <- vegdist(scale(transMetabData[c(5:156)]), "euclidean") #Create euclidean dissimilarity matrix to use for perMANOVA and PERMDISP
```

```{r}
permanova.metabid <- adonis2(dissim.metabid ~ treatment*day, data = transMetabData, method = 'eu', by = "terms") #Conduct global PERMANOVA to assess significance of trends in PCA. Scale metabolomics data and assess influence of treatment, day, and treatment x day on metabolite profiles
permanova.metabid #Significant influence of all factors
```

```{r}
broom::tidy(permanova.metabid) %>%
  write.csv(., "PCA/known-metabolites-PERMANOVA-results.csv", quote = FALSE) #Save PERMANOVA output
```

The fact that all three explanatory variables are significant means that the inclusion of unidentified metabolites doesn't drastically influence the results. I'm going to check these trends with a PERMDISP test as well.

```{r}
disp.metabid.temp <- betadisper(dissim.metabid, group = as.factor(transMetabData$treatment), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.metabid.temp) #Dispersion differences exist
boxplot(disp.metabid.temp) #Show distance to group centroids, can clearly see dispersion differences between 30 and the other two
anova(disp.metabid.temp) #Variance is different between groups. Temperature significance is due to centroid and dispersion differences!
```

```{r}
disp.metabid.day <- betadisper(dissim.metabid, group = as.factor(transMetabData$day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.metabid.day) #Similar dispersion, but some clear outliers for day 22
boxplot(disp.metabid.day) #Similar dispersion differences, outliers seen more easily
anova(disp.metabid.day) #Variance is different between groups. Day significance is due to centroid and dispersion differences!
```

```{r}
disp.metabid.tempDay <- betadisper(dissim.metabid, group = as.factor(transMetabData$treatment_day), type = 'centroid') #Run a beta dispersion model to assess if significant differences are due to differences in group centroid or variance
plot(disp.metabid.tempDay) #Dispersion differences exist, especially with 30_22
boxplot(disp.metabid.tempDay) #Very clear dispersion differences for each unique treatment!
anova(disp.metabid.tempDay) #Variance is different between groups. Interaction significance is due to centroid and dispersion differences!
```

```{r}
rbind(broom::tidy(anova(disp.metabid.temp)) %>%
        mutate(var = "temperature"),
      broom::tidy(anova(disp.metabid.day)) %>%
        mutate(var = "day"),
      broom::tidy(anova(disp.metabid.tempDay)) %>%
        mutate(var = "interaction")) %>%
  write.csv("PCA/known-metabolites-PERMDISP-results.csv", quote = FALSE, row.names = FALSE) #Combine PERMDISP output for each variable and add specifics for which variable was tested. Save output
```


# PLSDA for all metabolites

The next step is for me to conduct a PLS-DA analysis using the `mixOmics` package. This is a more robust way to analyze differences between treatments and is common in many metabolomics studies. Since both temperature and day were significant in the PCA, I'll use both of those factors in the PLSDA.

```{bash}
mkdir PLSDA
mkdir PLSDA/figures
```

## Temperature only

I'm still going to run the PLS-DA with temperature only because I suspect the plot will look cleaner this way.

```{r}
X <- log(transMetabData[c(5:601)]) #Assign only metabolite data to the X dataframe after log transformation.
Y <- as.factor(transMetabData$treatment) #Assign treatment data to the Y dataframe
```

```{r}
treatment.plsda <- plsda(X,Y, ncomp = 2) #ncomp = # treatments - 1
plotIndiv(treatment.plsda)
```

```{r}
#pdf("PLSDA/figures/all-data-PLSDA-treatment.pdf", height = 8.5, width = 11)
plotIndiv(treatment.plsda, comp = c(1,2), ind.names = FALSE,
          cex = 3, col = rev(plotColors), pch = c(15, 17, 19), point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC)", 
          title = "PLS-DA by Temperature", X.label = "Component 1: 14% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 6% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
pdf("PLSDA/figures/all-data-cimPlot-treatment.pdf", height = 8.5, width = 11)
cim(treatment.plsda, mapping = "Y", row.names = transMetabData$treatment, xlab = "Metabolites", ylab = "Temperature (ºC)", margins = c(8, 5), row.cex = 0.5) #Create CIM plot with metabolites on the x-axis and treatment on the y-axis. 
dev.off()
```

### Extract VIP

The VIP are the metabolites most important for distinguishing between treatments.

```{r}
treatmentVIP <- as.data.frame(PLSDA.VIP(treatment.plsda)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") #Extract VIP information and save as a dataframe. Convert metabolite rownames to a column. 
head(treatmentVIP) #Confirm changes
```

```{r}
write.csv(treatmentVIP, "PLSDA/all-data-VIP-treatment.csv", quote = FALSE, row.names = FALSE) #Save VIP scores
```

```{r}
treatmentVIP %>%
  filter(VIP >= 1) %>%
  nrow(.) #236 metabolites have VIP > 1
236/596 * 100 #39.6% metabolites VIP > 1
```

```{r}
treatmentVIP %>%
  filter(VIP >= 1) %>%
  arrange(VIP) %>%
  ggplot(aes(x = VIP, y = reorder(metabolite, VIP, sum))) +
  geom_point() +
  xlab("VIP Score") + ylab("Metabolite") +
  theme_classic() + theme(axis.text.y = element_text(size = 5)) #Using only metabolites with VIP > 1, arrange by metabolite name and reorder by cumulative VIP score. Plot points and change x and y axis labels.
ggsave("PLSDA/figures/all-data-VIP.pdf", height = 8.5, width = 11)
```
## Temperature x day

```{r}
X <- log(transMetabData[c(5:601)]) #Assign only metabolite data to the X dataframe after log transformation
Y <- as.factor(transMetabData$treatment_day) #Assign treatment data to the Y dataframe
```

```{r}
treatmentday.plsda <- plsda(X,Y, ncomp = 5) #ncomp = # treatments - 1
plotIndiv(treatmentday.plsda)
```
```{r}
#pdf("PLSDA/figures/all-data-PLSDA-day-treatment-comp12.pdf", height = 8.5, width = 11)
plotIndiv(treatmentday.plsda, comp = c(1,2), ind.names = FALSE,
          cex = 3, col = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"), pch = c(12:17), point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC) x Day", 
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 14% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 6% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
#pdf("PLSDA/figures/all-data-PLSDA-day-treatment-comp13.pdf", height = 8.5, width = 11)
plotIndiv(treatmentday.plsda, comp = c(1,3), ind.names = FALSE,
          cex = 3, col = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"), pch = c(12:17), point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC) x Day", 
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 14% Explained Variance", size.xlabel = 15, Y.label = "Component 3: 11% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
pdf("PLSDA/figures/all-data-cimPlot-day-treatment.pdf", height = 8.5, width = 11)
cim(treatment.plsda, mapping = "Y", row.names = transMetabData$treatment_day, xlab = "Metabolites", ylab = "Temperature (ºC) x Day", margins = c(8, 5), row.cex = 0.5) #Create CIM plot with metabolites on the x-axis and treatment x day on the y-axis. 
dev.off()
```

### Extract VIP

```{r}
treatmentdayVIP <- as.data.frame(PLSDA.VIP(treatmentday.plsda)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") #Extract VIP information and save as a dataframe. Convert metabolite rownames to a column.
head(treatmentdayVIP) #Confirm changes
```

```{r}
write.csv(treatmentdayVIP, "PLSDA/all-data-VIP-day-treatment.csv", quote = FALSE, row.names = FALSE) #Save VIP scores
```

```{r}
treatmentdayVIP %>%
  filter(VIP >= 1) %>%
  nrow(.) #251 metabolites have VIP > 1
251/596 * 100 #42.1% metabolites VIP > 1
```

```{r}
treatmentdayVIP %>%
  filter(VIP >= 1) %>%
  arrange(VIP) %>%
  ggplot(aes(x = VIP, y = reorder(metabolite, VIP, sum))) +
  geom_point() +
  xlab("VIP Score") + ylab("Metabolite") +
  theme_classic() + theme(axis.text.y = element_text(size = 5)) #Using only metabolites with VIP > 1, arrange by metabolite name and reorder by cumulative VIP score. Plot points and change x and y axis labels.
ggsave("PLSDA/figures/all-data-VIP-day-treatment.pdf", height = 8.5, width = 11)
```

# PLSDA for known features

While I can use the PLSDA with all features to discuss larger patterns and maybe glean insight into functions of unknown metabolites, running the model with only known features will allow me to more confidently assess differences in metabolic pathways due to temperature and time. I will run this PLSDA for both temperature and day.

```{r}
X <- log(transMetabData[c(5:156)]) #Assign known metabolite data to the X dataframe after log transformation
Y <- as.factor(transMetabData$treatment_day) #Assign treatment data to the Y dataframe
```

```{r}
treatmentday.known.plsda <- plsda(X,Y, ncomp = 5) #ncomp = # treatments - 1
plotIndiv(treatmentday.known.plsda)
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-day-treatment-comp12.pdf", height = 8.5, width = 11)
plotIndiv(treatmentday.known.plsda, comp = c(1,2), ind.names = FALSE,
          cex = 3, col = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"), pch = rep(16, times = 6), point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC) x Day",
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 12% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 12% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```
```{r}
#svg("PLSDA/figures/known-metabolites-PLSDA-day-treatment-comp12.svg", height = 8.5, width = 11)
plotIndiv(treatmentday.known.plsda, comp = c(1,2), ind.names = FALSE,
          cex = 3, col = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"), pch = rep(16, times = 6), point.lwd = 0.7,
          ellipse = FALSE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC) x Day", legend.position = "bottom",
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 12% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 12% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
#pdf("PLSDA/figures/known-metabolites-PLSDA-day-treatment-comp13.pdf", height = 8.5, width = 11)
plotIndiv(treatmentday.known.plsda, comp = c(1,3), ind.names = FALSE,
          cex = 3, col = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"), pch = c(12:17), point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC) x Day", 
          title = "PLS-DA by Temperature and Day", X.label = "Component 1: 12% Explained Variance", size.xlabel = 15, Y.label = "Component 3: 8% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2") #Visualize first two components of PLS-DA with no individual sample names. Alter point size, color, shape, and line width. Include 95% confidence ellipses and use star format to visualize sample distance from centroids. Include a legend and specify the legend title. Specify the plot title and text label sizes. Use ggplot style.
#dev.off()
```

```{r}
pdf("PLSDA/figures/known-metabolites-cimPlot-day-treatment.pdf", height = 8.5, width = 11)
cim(treatmentday.known.plsda, mapping = "Y", row.names = transMetabData$treatment_day, xlab = "Metabolites", ylab = "Temperature (ºC) x Day", margins = c(8, 5), row.cex = 0.5) #Create CIM plot with metabolites on the x-axis and treatment x day on the y-axis. 
dev.off()
```

### Extract VIP

```{r}
treatmentday.known.VIP <- as.data.frame(PLSDA.VIP(treatmentday.known.plsda)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") #Extract VIP information and save as a dataframe. Convert metabolite rownames to a column.
head(treatmentdayVIP) #Confirm changes
```

```{r}
write.csv(treatmentday.known.VIP, "PLSDA/known-metabolites-VIP-day-treatment.csv", quote = FALSE, row.names = FALSE) #Save VIP scores
```

```{r}
treatmentday.known.VIP %>%
  filter(VIP >= 1) %>%
  nrow(.) #73 metabolites have VIP > 1
73/151 * 100 #48.3% metabolites VIP > 1
```

```{r}
treatmentday.known.VIP %>%
  filter(VIP >= 1) %>%
  arrange(VIP) %>%
  ggplot(aes(x = VIP, y = reorder(metabolite, VIP, sum))) +
  geom_point() +
  xlab("VIP Score") + ylab("Metabolite") +
  theme_classic() + theme(axis.text.y = element_text(size = 8)) #Using only metabolites with VIP > 1, arrange by metabolite name and reorder by cumulative VIP score. Plot points and change x and y axis labels.
ggsave("PLSDA/figures/known-metabolites-VIP-day-treatment.pdf", height = 8.5, width = 11)
```

```{r}
significantVIP <- treatmentday.known.VIP %>%
  filter(VIP >= 1) #Save significant VIP
```

## Pairwise tests for VIP metabolites

We now know which metabolites are most important for separating days and treatments. However, we do not know specifically which VIP distinguish between which treatment x day contrasts. I will use pairwise t-tests to identify which VIP are important for which contrasts. 

### Treatment differences at day 3

#### 13_3 vs. 30_3

```{r}
list <- c("13_3", "30_3") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_13v30_day3 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_13v30_day3 <- as.data.frame(PLSDA.VIP(known.plsda_13v30_day3)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
```

```{r}
clean_13v30_day3 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_13v30_day3 <- cbind(clean_13v30_day3[, 1:4], 
                               log(clean_13v30_day3[, names(clean_13v30_day3) %in% VIP_13v30_day3$metabolite])) %>%
  pivot_longer(., cols = 5:63, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_13v30_day3 #Confirm changes
```

```{r}
t.test_13v30_day3 <-do(VIP_select_13v30_day3, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                          alternative = "two.sided",
                                                          mu = 0,
                                                          paired = FALSE,
                                                          var.equal = FALSE,
                                                          conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_13v30_day3$p.adj <- p.adjust(t.test_13v30_day3$p.value, method = c("fdr"), n = length(VIP_13v30_day3$metabolite)) #Adjust p value for the number of comparisons
t.test_13v30_day3
```

```{r}
t.test_13v30_day3 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

#### 13_3 vs. 5_3

```{r}
list <- c("13_3", "5_3") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_13v5_day3 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_13v5_day3 <- as.data.frame(PLSDA.VIP(known.plsda_13v5_day3)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_13v5_day3
```

```{r}
clean_13v5_day3 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_13v5_day3 <- cbind(clean_13v5_day3[, 1:4], 
                              log(clean_13v5_day3[, names(clean_13v5_day3) %in% VIP_13v5_day3$metabolite])) %>%
  pivot_longer(., cols = 5:66, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_13v5_day3 #Confirm changes
```

```{r}
t.test_13v5_day3 <-do(VIP_select_13v5_day3, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        paired = FALSE,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_13v5_day3$p.adj <- p.adjust(t.test_13v5_day3$p.value, method = c("fdr"), n = length(VIP_13v5_day3$metabolite)) #Adjust p value for the number of comparisons
t.test_13v5_day3
```

```{r}
t.test_13v5_day3 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

#### 30_3 vs. 5_3

```{r}
list <- c("30_3", "5_3") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_30v5_day3 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_30v5_day3 <- as.data.frame(PLSDA.VIP(known.plsda_30v5_day3)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_30v5_day3
```

```{r}
clean_30v5_day3 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_30v5_day3 <- cbind(clean_30v5_day3[, 1:4], 
                              log(clean_30v5_day3[, names(clean_30v5_day3) %in% VIP_30v5_day3$metabolite])) %>%
  pivot_longer(., cols = 5:69, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_30v5_day3 #Confirm changes
```

```{r}
t.test_30v5_day3 <-do(VIP_select_30v5_day3, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_30v5_day3$p.adj <- p.adjust(t.test_30v5_day3$p.value, method = c("fdr"), n = length(VIP_30v5_day3$metabolite)) #Adjust p value for the number of comparisons
t.test_30v5_day3
```

```{r}
t.test_30v5_day3 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #9 significantly different metabolites
t.test_30v5_day3 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #9 metabolites higher in 30ºC
t.test_30v5_day3 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #0 metabolites higher in 5ºC
```

```{r}
t.test_30v5_day3 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_30v5_day3, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "30",
                                  estimate < 0 ~ "5")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c(plotColors[1], plotColors[3]), 
                    name = "Temperature (ºC)",
                    breaks = c("30", "5"),
                    labels = c("30", "5")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-30v5-day3.pdf", height = 8.5, width = 11)
```

### Treatment differences at day 22

#### 13_22 vs. 30_22

```{r}
list <- c("13_22", "30_22") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_13v30_day22 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_13v30_day22 <- as.data.frame(PLSDA.VIP(known.plsda_13v30_day22)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_13v30_day22
```

```{r}
clean_13v30_day22 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_13v30_day22 <- cbind(clean_13v30_day22[, 1:4], 
                                log(clean_13v30_day22[, names(clean_13v30_day22) %in% VIP_13v30_day22$metabolite])) %>%
  pivot_longer(., cols = 5:55, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_13v30_day22 #Confirm changes
```

```{r}
t.test_13v30_day22 <-do(VIP_select_13v30_day22, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                            alternative = "two.sided",
                                                            mu = 0,
                                                            paired = FALSE,
                                                            var.equal = FALSE,
                                                            conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_13v30_day22$p.adj <- p.adjust(t.test_13v30_day22$p.value, method = c("fdr"), n = length(VIP_13v30_day22$metabolite)) #Adjust p value for the number of comparisons
t.test_13v30_day22
```

```{r}
t.test_13v30_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #29 significantly different metabolites
t.test_13v30_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #16 metabolites higher in 13ºC
t.test_13v30_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #13 metabolites higher in 30ºC
```

```{r}
t.test_13v30_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_13v30_day22, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "13",
                                  estimate < 0 ~ "30")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[1]), 
                    name = "Temperature (ºC)",
                    breaks = c("13", "30"),
                    labels = c("13", "30")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-13v30-day22.pdf", height = 8.5, width = 11)
```

#### 13_22 vs. 5_22

```{r}
list <- c("13_22", "5_22") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_13v5_day22 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_13v5_day22 <- as.data.frame(PLSDA.VIP(known.plsda_13v5_day22)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_13v5_day22
```

```{r}
clean_13v5_day22 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_13v5_day22 <- cbind(clean_13v5_day22[, 1:4], 
                               log(clean_13v5_day22[, names(clean_13v5_day22) %in% VIP_13v5_day22$metabolite])) %>%
  pivot_longer(., cols = 5:46, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_13v5_day22 #Confirm changes
```

```{r}
t.test_13v5_day22 <-do(VIP_select_13v5_day22, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                          alternative = "two.sided",
                                                          mu = 0,
                                                          paired = FALSE,
                                                          var.equal = FALSE,
                                                          conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_13v5_day22$p.adj <- p.adjust(t.test_13v5_day22$p.value, method = c("fdr"), n = length(VIP_13v5_day22$metabolite)) #Adjust p value for the number of comparisons
t.test_13v5_day22
```

```{r}
t.test_13v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #31 significantly different metabolites
t.test_13v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #29 metabolites higher in 13ºC
t.test_13v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #2 metabolites higher in 5ºC
```

```{r}
t.test_13v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_13v5_day22, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "13",
                                  estimate < 0 ~ "5")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[3]), 
                    name = "Temperature (ºC)",
                    breaks = c("13", "5"),
                    labels = c("13", "5")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-13v5-day22.pdf", height = 8.5, width = 11)
```

#### 30_22 vs. 5_22

```{r}
list <- c("30_22", "5_22") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_30v5_day22 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_30v5_day22 <- as.data.frame(PLSDA.VIP(known.plsda_30v5_day22)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_30v5_day22
```

```{r}
clean_30v5_day22 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_30v5_day22 <- cbind(clean_30v5_day22[, 1:4], 
                               log(clean_30v5_day22[, names(clean_30v5_day22) %in% VIP_30v5_day22$metabolite])) %>%
  pivot_longer(., cols = 5:70, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_30v5_day22 #Confirm changes
```

```{r}
t.test_30v5_day22 <-do(VIP_select_30v5_day22, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                          alternative = "two.sided",
                                                          mu = 0,
                                                          var.equal = FALSE,
                                                          conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_30v5_day22$p.adj <- p.adjust(t.test_30v5_day22$p.value, method = c("fdr"), n = length(VIP_30v5_day22$metabolite)) #Adjust p value for the number of comparisons
t.test_30v5_day22
```

```{r}
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #42 significantly different metabolites
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #10 metabolites higher in 30ºC
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #32 metabolites higher in 5ºC
```

```{r}
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_30v5_day22, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "30",
                                  estimate < 0 ~ "5")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c(plotColors[1], plotColors[3]), 
                    name = "Temperature (ºC)",
                    breaks = c("30", "5"),
                    labels = c("30", "5")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("../output/05-metabolomics-analysis/PLSDA/figures/significant-VIP-30v5-day22.pdf", height = 8.5, width = 11)
```

Since I have identified VIP for 13ºC vs. either 30ºC or 5ºC at day 22, I want to see if the 30 vs. 5 VIP are unique to this contrast or not.

```{r}
#Take the t-test results and filter out the significant entries. Use anti-join to identify metabolites that are unique to the 30 vs. 5 comparison.

t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  anti_join(x = ., y = (t.test_13v5_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  anti_join(x = ., y = (t.test_13v30_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  nrow(.) # 15 significantly different metabolites
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  anti_join(x = ., y = (t.test_13v5_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  anti_join(x = ., y = (t.test_13v30_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #1 metabolites higher in 30ºC
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  anti_join(x = ., y = (t.test_13v5_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  anti_join(x = ., y = (t.test_13v30_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #14 metabolites higher in 5ºC
```

```{r}
t.test_30v5_day22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  anti_join(x = ., y = (t.test_13v5_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  anti_join(x = ., y = (t.test_13v30_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  left_join(., VIP_30v5_day22, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "30",
                                  estimate < 0 ~ "5")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("metabolite") +
  scale_fill_manual(values = c(plotColors[1], plotColors[3]), 
                    name = "Temperature (ºC)",
                    breaks = c("30", "5"),
                    labels = c("30", "5")) +
  theme_classic(base_size = 15) #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by identifier name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/unique-significant-VIP-30v5-day22.pdf.pdf", height = 8.5, width = 11)
```

### Day differences for each treatment

#### 5_3 vs. 5_22

```{r}
list <- c("5_3", "5_22") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_5_day3v22 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_5_day3v22 <- as.data.frame(PLSDA.VIP(known.plsda_5_day3v22)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_5_day3v22
```

```{r}
clean_5_day3v22 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_5_day3v22 <- cbind(clean_5_day3v22[, 1:4], 
                              log(clean_5_day3v22[, names(clean_5_day3v22) %in% VIP_5_day3v22$metabolite])) %>%
  pivot_longer(., cols = 5:55, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_5_day3v22 #Confirm changes
```

```{r}
t.test_5_day3v22 <-do(VIP_select_5_day3v22, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                        alternative = "two.sided",
                                                        mu = 0,
                                                        paired = FALSE,
                                                        var.equal = FALSE,
                                                        conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_5_day3v22$p.adj <- p.adjust(t.test_5_day3v22$p.value, method = c("fdr"), n = length(VIP_5_day3v22$metabolite)) #Adjust p value for the number of comparisons
t.test_5_day3v22
```

```{r}
t.test_5_day3v22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #6 significantly different metabolites
t.test_5_day3v22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 > estimate2) %>%
  nrow(.) #0 metabolites higher in day 22
t.test_5_day3v22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>% filter(., estimate1 < estimate2) %>%
  nrow(.) #6 metabolites higher in day 3
```

```{r}
t.test_5_day3v22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  left_join(., VIP_5_day3v22, by = "metabolite") %>%
  mutate(., direction = case_when(estimate > 0 ~ "22",
                                  estimate < 0 ~ "3")) %>%
  mutate(., score = case_when(estimate > 0 ~ VIP,
                              estimate < 0 ~ -VIP)) %>%
  arrange(score) %>%
  ggplot(aes(x = score, y = reorder(metabolite, score, sum))) +
  geom_bar(aes(fill = direction), stat = 'identity') +
  xlab("VIP Score") + ylab("Metabolite") +
  scale_fill_manual(values = c("black", "purple"), 
                    name = "Day",
                    breaks = c("3", "22"),
                    labels = c("3", "22")) +
  theme_classic(base_size = 15) + theme() #Take t-test results and filter out significant metabolites. Create two new columns using estimate values: one for the direction of increase, and another for positive or negative VIP. Arrange by metabolite name and reorder by cumulative VIP score. Plot bars and change x and y axis labels. Add scale information.
ggsave("PLSDA/figures/significant-VIP-5-day3v22.pdf", height = 8.5, width = 11)
```

#### 13_3 vs. 13_22

```{r}
list <- c("13_3", "13_22") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_13_day3v22 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_13_day3v22 <- as.data.frame(PLSDA.VIP(known.plsda_13_day3v22)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_13_day3v22
```

```{r}
clean_13_day3v22 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_13_day3v22 <- cbind(clean_13_day3v22[, 1:4], 
                               log(clean_13_day3v22[, names(clean_13_day3v22) %in% VIP_13_day3v22$metabolite])) %>%
  pivot_longer(., cols = 5:61, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_13_day3v22 #Confirm changes
```

```{r}
t.test_13_day3v22 <-do(VIP_select_13_day3v22, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                          alternative = "two.sided",
                                                          mu = 0,
                                                          paired = FALSE,
                                                          var.equal = FALSE,
                                                          conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_13_day3v22$p.adj <- p.adjust(t.test_13_day3v22$p.value, method = c("fdr"), n = length(VIP_13_day3v22$metabolite)) #Adjust p value for the number of comparisons
t.test_13_day3v22
```

```{r}
t.test_13_day3v22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

#### 30_3 vs. 30_22

```{r}
list <- c("30_3", "30_22") #Assign treatments to examine
```

```{r}
X <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts
Y <- as.factor(X$treatment_day) #Treatments for Y
X <- X[, 5:156] #Retain data only
```

```{r}
known.plsda_30_day3v22 <- plsda(X, Y, ncomp = 5) #Run PLSDA for desired contrast
VIP_30_day3v22 <- as.data.frame(PLSDA.VIP(known.plsda_30_day3v22)[["tab"]]) %>%
  rownames_to_column(., var = "metabolite") %>%
  filter(., VIP >= 1) #Extract VIP list and filter for VIP > 1
VIP_30_day3v22
```

```{r}
clean_30_day3v22 <- transMetabData %>%
  dplyr::filter(., treatment_day %in% list) %>%
  droplevels() #Filter transMetabData for desired contrasts.
VIP_select_30_day3v22 <- cbind(clean_30_day3v22[, 1:4], 
                               log(clean_30_day3v22[, names(clean_30_day3v22) %in% VIP_30_day3v22$metabolite])) %>%
  pivot_longer(., cols = 5:67, values_to = "log_norm_counts", names_to = "metabolite") %>%
  group_by(., metabolite) #cbind metadata columns (1:4) and log-transformed "clean" data for VIP. Pivot data longer and group by metabolites.
VIP_select_30_day3v22 #Confirm changes
```

```{r}
t.test_30_day3v22 <-do(VIP_select_30_day3v22, tidy(t.test(.$log_norm_counts ~ .$treatment_day,
                                                          alternative = "two.sided",
                                                          mu = 0,
                                                          paired = FALSE,
                                                          var.equal = FALSE,
                                                          conf.level = 0.95
))) #Looped t-test for all metabolites with VIP > 1
t.test_30_day3v22$p.adj <- p.adjust(t.test_30_day3v22$p.value, method = c("fdr"), n = length(VIP_30_day3v22$metabolite)) #Adjust p value for the number of comparisons
t.test_30_day3v22
```

```{r}
t.test_30_day3v22 %>%
  ungroup(.) %>%
  filter(., p.adj < 0.05) %>%
  nrow(.) #0 significantly different metabolites
```

# Validate PLSDA model

It's important to validate the PLSDA model. First, I'll run a permutation test to determine how many components to include.

```{r}
MyPerf.plsda <- perf(treatmentday.known.plsda, validation = "Mfold", folds = 6,  
                     progressBar = TRUE, nrepeat = 50) 

plot(MyPerf.plsda, sd = TRUE) #Overall, increasing the number of components decreases the error rate. There doesn't seem to be an exponential reduction of error rate after 3 components.
```

Next, I'll assess the optimal number of metabolites to keep in the analysis.

```{r}
list.keepX <- c(5:10,  seq(20, 300, 10)) #list.keepX # to output the grid of values tested
X <- log(transMetabData[c(5:156)]) #Assign known metabolite data to the X dataframe after log transformation
Y <- as.factor(transMetabData$treatment_day) #Assign treatment data to the Y dataframe
```

```{r}
tune.splsda.srbct <- tune.splsda(X, Y, ncomp = 5, 
                                 validation = 'Mfold', folds = 6,
                                 measure = "BER", test.keepX = list.keepX, nrepeat = 50) #More extensive classification performance
plot(tune.splsda.srbct) #Plot results
```

```{r}
error <- tune.splsda.srbct$error.rate #Save error rate
ncomp <- tune.splsda.srbct$choice.ncomp$ncomp #Optimal number of components based on t-tests on the error rate
ncomp #5 components! That's how many I have

select.keepX <- tune.splsda.srbct$choice.keepX #Optimal number of variables to select (e.g., number of metabolite features)
select.keepX #sPLSDA should be run with 40 features
```

```{r}
myResult.splsda.final_T <- splsda(X, Y, ncomp = ncomp, keepX = select.keepX) #Use the number of components and meatbolites to keep based on optimization from above  
```

```{r}
#pdf("PLSDA/figures/known-metabolites-sPLSDA-day-treatment-comp12.pdf", height = 8.5, width = 11)
plotIndiv(myResult.splsda.final_T, comp = c(1,2), ind.names = FALSE,
          cex = 3, col = c("#525252", "grey80", "#CB181D", "salmon", "#2171B5", "lightblue"), pch = c(12:17), point.lwd = 0.7,
          ellipse = TRUE, star = TRUE, centroid = FALSE, 
          legend = TRUE, legend.title = "Temperature (ºC) x Day", 
          title = "sPLS-DA by Temperature and Day", X.label = "Component 1: 9% Explained Variance", size.xlabel = 15, Y.label = "Component 2: 6% Explained Variance", size.ylabel = 15, size.axis = 13,
          style = "ggplot2")
#dev.off()
```

# WCNA analysis for known metabolites

Although I have lists of VIP associated with each pairwise comparison, there is one additional step I can take to determine which known metabolites are associated with different treatment x day combinations. I will use a Weighted Correlation Network Analysis (similar to a WGCNA for gene expression) to group metabolites together by abundance patterns, then correlate metabolite modules with treatment and day.

```{bash}
mkdir WCNA
mkdir WCNA/figures
```

## Data preparation

### Format data

I need to convert my dataframe such that samples are columns and metabolites are rows. I also need to have a dataframe without metadata.

```{r}
head(rawMetabolomicsData)
head(metabolomicsMetadata)
```

```{r}
rowwiseMetabData <- rawMetabolomicsData %>%
  dplyr::rename(., metabolite = BinBase.name) %>%
  dplyr::select(., -c(2:8)) %>%
  column_to_rownames(., var = "metabolite") %>%
  slice(., c(1:152)) %>%
  log(.) #Take the raw metabolomics data, since that dataframe already has samples in columns and metabolites in rows. Rename the metabolite column, get rid of extraneous data, and change the metabolite column to rownames. Keep only data from known metabolites. Log transform the data.
tail(rowwiseMetabData) #Confirm formatting
```

```{r}
rowSums(dplyr::count(rowwiseMetabData)) > 0 #Check that there are no metabolites with 0 counts for all samples. Should return TRUE.  
```

### Data filtering: PoverA and genefilter

*pOverA*: Specifying the minimum count for a proportion of samples for each metabolite. Here, we are using a pOverA of 0.08. This is because we have 69 samples with a minimum of n = 6 samples per treatment x day. Therefore, we will accept metabolites that are present in 6/69 = 0.08 of the samples because we expect different metabolites by temperature and day as demonstrated by PLSDA analysis. We are further setting the minimum value of metabolites to 0.01 (median normalized), such that 8% of the samples must have a non-zero normalized metabolite presence in order for the metabolite to remain in the data set.  

Filter in the package "genefilter". Pre-filtering our dataset to reduce the memory size dataframe, increase the speed of the transformation and testing functions, and improve quality of statistical analysis by removing low-coverage counts. Removed counts could represent outliers in the data and removing these improves sensitivity of statistical tests.

```{r}
filt <- filterfun(pOverA(0.08, 0.01)) #Define the filter
gfilt <- genefilter(rowwiseMetabData, filt) #Create filter for the metabolite data
```

```{r}
keep <- rowwiseMetabData[gfilt,] #Identify metabolites to keep by count filter
n.keep <- rownames(keep) #Identify metabolite lists

rowwiseMetabData_filt <- as.data.frame(rowwiseMetabData[which(rownames(rowwiseMetabData) %in% n.keep),]) #Metabolite count data filtered in PoverA, P percent of the samples have counts over A
```

```{r}
#How many rows do we have before and after filtering?
nrow(rowwiseMetabData) #Before
nrow(rowwiseMetabData_filt) #After
```

All 152 metabolites will be used in downstream analysis.

```{r}
#Checking that all row and column names match. Should return "TRUE"
all(rownames(metabolomicsMetadata$sample_id) %in% colnames(rowwiseMetabData_filt))
all(rownames(metabolomicsMetadata$sample_id) == colnames(rowwiseMetabData_filt)) 
```

### Outlier detection

I will identify outliers visually through a clustering analysis.

```{r}
sampleTree <- hclust(dist(rowwiseMetabData), method = "average") #Perform clustering analysis
```

```{r}
#pdf("WCNA/figures/metabolite-outliers.pdf", height = 8.5, width = 11)
plot(sampleTree, main = "Sample clustering to detect outliers", sub = "", xlab = "", cex.lab = 0.8, cex.axis = 0.8, cex.main = 2) #Plot cluster dendogram
#dev.off()
```

Based on the tree, taurine and sulfuric acid might be outliers.

```{r}
rowwiseMetabData %>% filter(., rownames(.) == "taurine")
rowwiseMetabData %>% filter(., rownames(.) == "sulfuric acid")
```

```{r}
rowwiseMetabData <- t(rowwiseMetabData) %>% as.data.frame(.) #Transpose such that samples are in rows and metabolites are in columns.  
head(rowwiseMetabData) #Confirm changes
```

Construct a new cluster dendogram to visually identify sample outliers.

```{r}
sampleTree <- hclust(dist(rowwiseMetabData), method = "average") #Perform clustering analysis
```

```{r}
#pdf("WCNA/figures/sample-outliers.pdf", height = 8.5, width = 11)
plot(sampleTree, main = "Sample clustering to detect outliers", sub = "", xlab = "", cex.lab = 1.5, cex.axis = 1.5, cex.main = 2)
#dev.off()
```

There don't seem to be any samples that look like outliers (but it's also interesting how there isn't a clear separation of samples by tank or treatment).

## Network construction and consensus module detection

### Choosing a soft-thresholding power: Analysis of a network topology β  

The soft thresholding power (β) is the number to which the co-expression similarity is raised to calculate adjacency. The function `pickSoftThreshold` performs a network topology analysis. The user chooses a set of candidate powers, however the default parameters are suitable values.  

```{r, message=FALSE, warning=FALSE}
allowWGCNAThreads()
powers <- c(c(1:20), seq(from = 12, to=20, by=2)) #Create a string of numbers from 1 through 10, and even numbers from 10 through 20
sft <-pickSoftThreshold(rowwiseMetabData, powerVector = powers, verbose = 5) # Call the network topology analysis function
```

```{r}
sizeGrWindow(9, 5)
par(mfrow = c(1,2));
cex1 = 0.9;
# # # Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"));
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red");
# # # this line corresponds to using an R^2 cut-off
abline(h=0.85,col="red")
# # # Mean connectivity as a function of the soft-thresholding power
plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```

When choosing a soft thresholding power, want to choose the lowest number that reaches the R<sup>2</sup> cutoff. From this data, it appears that the **soft thresholding power is 6**.  

### Identify modules using `blockwiseModules`  

Use `blockwiseModules` to identify modules of metabolites.  

Settings used: 

networkType = "unsigned" 
deepSplit = 2
pamRespectsDendro = F
minModuleSize = 5
maxBlockSize = 4000
reassignThreshold = 0
mergeCutHeight = 0.25

```{r, echo=TRUE, warning=FALSE, message=FALSE}

picked_power = 6
temp_cor <- cor       
cor <- WGCNA::cor # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(rowwiseMetabData,                         # <= input here
                          
                          # == Adjacency Function ==
                          power = picked_power,               # <= power here
                          networkType = "unsigned",
                          
                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 5,                  
                          maxBlockSize = 4000,
                          
                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,
                          
                          # == TOM == Archive the run results in TOM file (saves time) but it doesn't save a file
                          saveTOMs = F,
                          saveTOMFileBase = "ER",
                          
                          # == Output Options
                          numericLabels = T,
                          verbose = 3)

cor <- temp_cor     # Return cor function to original namespace
```

```{r}
mergedColors = netwk$colors # Identify labels as numbers 
table(mergedColors)
```

Module sizes are as follows: 

0  1  2  3  4  5  6  7  8 
40 22 22 19 15 10 10  8  6 

```{r}
# Plot the dendrogram and the module colors underneath

#pdf("WCNA/figures/cluster-dendogram-mergedME.pdf")
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
#dev.off()
```

### Relate modules to sample information

``` {r, echo=TRUE, warning=FALSE, message=FALSE}
module_df <- data.frame(
  Metabolite = names(netwk$colors),
  colors = netwk$colors
  #colors = labels2colors(netwk$colors)
)

module_df[1:5,]

write.csv(module_df, "WCNA/metabolite_modules.csv")

# Get Module Eigengenes per cluster
MEs <- moduleEigengenes(rowwiseMetabData, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs <- orderMEs(MEs)
module_order = names(MEs) %>% gsub("ME","", .)

# Add Sample names
MEs0 <- MEs
MEs0$crab.ID = row.names(MEs)

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-crab.ID) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  )

mME %>% ggplot(., aes(x=crab.ID, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_gradient2(
    low = "blue",
    high = "red",
    mid = "white",
    midpoint = 0,
    limit = c(-1,1)) +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-Sample Relationships", y = "Modules", fill="corr")
```

### Relate modules to treatment and day

Prepare trait data. Data has to be numeric, so I will substitute temperatures and days for numeric values. TMake a dataframe that has a column for each treatment or day and a row for samples. Populate a 1 for samples that match each treatment or day and a 0 for samples not matching respective treatments or days This process changes treatments and days from a categorical variable into a binary variable. This will allow for correlations between mean eigengenes and treatment or day.  

```{r}
head(metabolomicsMetadata)
```

```{r}
datTraits <- metabolomicsMetadata %>%
  mutate(., "13ºC at Day 3" = case_when(treatment_day == "13_3" ~ 1,
                                        treatment_day != "13_3" ~ 0)) %>%
  mutate(., "5ºC at Day 3" = case_when(treatment_day == "5_3" ~ 1,
                                       treatment_day != "5_3" ~ 0)) %>%
  mutate(., "30ºC at Day 3" = case_when(treatment_day == "30_3" ~ 1,
                                        treatment_day != "30_3" ~ 0)) %>%
  mutate(., "13ºC at Day 22" = case_when(treatment_day == "13_22" ~ 1,
                                         treatment_day != "13_22" ~ 0)) %>%
  mutate(., "5ºC at Day 22" = case_when(treatment_day == "5_22" ~ 1,
                                        treatment_day != "5_22" ~ 0)) %>%
  mutate(., "30ºC at Day 22" = case_when(treatment_day == "30_22" ~ 1,
                                         treatment_day != "30_22" ~ 0)) %>%
  dplyr::select(., -c(2:4)) #Copy dataframe into datTraits, then use a series of case_when commands to create individual columns for each treatment or day, with 1 indicating that sample is part of the group, and 0 indicating that sample is not part of the group. Remove previous treatment classification columns. 
rownames(datTraits) <- datTraits$crab.ID #Convert crab.ID into rownames.
datTraits <- datTraits %>%
  dplyr::select(., -crab.ID) #Remove the crab.ID column
head(datTraits) #Confirm dataframe creation
```

```{r}
#Define numbers of metabolites and samples and view 
nMetabolites <- ncol(rowwiseMetabData)
nSamples <- nrow(rowwiseMetabData)

nMetabolites
nSamples
```

We have 152 metabolites and 69 samples.   

Next correlate the trait matrix (treatment, day) with eigengenes (modules).  

```{r}
moduleTraitCor <- cor(MEs, datTraits, use = "p") #Correlations of traits with eigengenes
moduleTraitPvalue <- corPvalueStudent(moduleTraitCor, nSamples) #Get p-values for correlations
head(moduleTraitCor %>% as.data.frame())
head(moduleTraitPvalue %>% as.data.frame())
```

```{r}
moduleTraitTree <- hclust(dist(t(moduleTraitCor)), method = "average") #Create cluster dendogram for module-trait correlations
```

```{r}
#pdf(file = "WCNA/figures/module-trait-cluster-dendogram.pdf", height = 8.5, width = 11)
plot(moduleTraitTree) #Plot cluster dendogram
#dev.off()
```

```{r}
# Correlations of metabolites with eigengenes. Calculate correlations between ME's and groups 
moduleGeneCor <- cor(MEs, rowwiseMetabData) #Correlate eigengenes with metabolites
moduleGenePvalue <- corPvalueStudent(moduleGeneCor, nSamples) #Calculate p-values for metabolite-eigengene correlations
head(moduleGeneCor %>% as.data.frame)
head(moduleGenePvalue %>% as.data.frame(.))
```

## Plot module-trait associations

```{r, echo=TRUE, warning=FALSE, message=FALSE}
textMatrix <- paste(signif(moduleTraitCor, 2), "\n(",signif(moduleTraitPvalue, 1), ")", sep = "") #Create a matrix with the module-trait correlation and p-values
dim(textMatrix) <- dim(moduleTraitCor) #Ensure this matrix has the appropriate dimensions
head(textMatrix) #Confirm matrix creation
```

```{r}
#pdf("WCNA/figures/module-trait-relationships.pdf", height = 8.5, width = 11)
labeledHeatmap(Matrix = moduleTraitCor, xLabels = names(datTraits),  yLabels = names(MEs), ySymbols = names(MEs), cex.lab.y = 1.5, cex.lab.x = 1.5, colors = blueWhiteRed(50), textMatrix = textMatrix, setStdMargins = TRUE, cex.text = 1, zlim = c(-1,1), main = paste("Module-trait relationships"), cex.main = 2) #Use the textMatrix to create a heatmap with correlations between modules and the various treatments.
#dev.off()
```

```{r, echo=TRUE, warning=FALSE, message=FALSE}
heatmappval <- signif(moduleTraitPvalue, 1) #Create list of pvalues for eigengene correlation with specific traits

row_dend <- dendsort(hclust(dist(moduleTraitCor))) #Create dendograms for rows
col_dend <- dendsort(hclust(dist(t(moduleTraitCor)))) #Create dendograms for columns

trait_order <- c("5ºC at Day 3", "13ºC at Day 3", "30ºC at Day 3", "5ºC at Day 22", "13ºC at Day 22", "30ºC at Day 22") #Order of traits for heatmap
```

```{r}
#pdf(file = "WCNA/figures/module-trait-relationship-heatmap.pdf", height = 8.5, width = 8.5)
ht <- Heatmap(moduleTraitCor, name = "Eigengene", column_title = "Module-Group Eigengene Correlation",
              col = blueWhiteRed(50), 
              row_names_side = "left", 
              row_dend_side = "left",
              width = unit(5, "in"), 
              height = unit(4.5, "in"), 
              column_dend_reorder = TRUE, 
              cluster_columns = col_dend,
              row_dend_reorder = FALSE,  
              #column_split = 3, 
              #row_split = 3, 
              #column_dend_height = unit(.5, "in"),
              cluster_rows = row_dend, 
              #column_order = trait_order, 
              #row_gap = unit(2.5, "mm"), 
              border = TRUE,
              cell_fun = function(j, i, x, y, w, h, col) {
                if(heatmappval[i, j] < 0.05) {
                  grid.text(sprintf("%s", heatmappval[i, j]), x, y, gp = gpar(fontsize = 10, fontface = "bold"))
                }
                else {
                  grid.text(sprintf("%s", heatmappval[i, j]), x, y, gp = gpar(fontsize = 10, fontface = "plain"))
                }},
              column_names_rot = 45,
              column_names_gp =  gpar(fontsize = 12, border = FALSE),
              column_title_gp = gpar(fontsize = 20),
              row_names_gp = gpar(fontsize = 12, alpha = 0.75, border = FALSE)) #Create a heatmap object
draw(ht) #Draw heatmap
#dev.off()
```

```{r}
#svg(file = "WCNA/figures/module-trait-relationship-heatmap.svg", height = 8.5, width = 8.5)
ht <- Heatmap(moduleTraitCor, name = "Correlation", column_title = "Module-Group Eigengene Correlation",
              row_names_side = "left", 
              row_dend_side = "left",
              width = unit(5, "in"), 
              height = unit(4.5, "in"), 
              column_dend_reorder = TRUE, 
              cluster_columns = col_dend,
              row_dend_reorder = FALSE,  
              #column_split = 3, 
              #row_split = 3, 
              #column_dend_height = unit(.5, "in"),
              cluster_rows = row_dend, 
              #column_order = trait_order, 
              #row_gap = unit(2.5, "mm"), 
              border = TRUE,
              cell_fun = function(j, i, x, y, w, h, col) {
                if(heatmappval[i, j] < 0.05) {
                  grid.text(sprintf("%s", heatmappval[i, j]), x, y, gp = gpar(fontsize = 10, fontface = "bold"))
                }
                else {
                  grid.text(sprintf("%s", heatmappval[i, j]), x, y, gp = gpar(fontsize = 10, fontface = "plain"))
                }},
              column_names_rot = 45,
              column_names_gp =  gpar(fontsize = 12, border = FALSE),
              column_title_gp = gpar(fontsize = 20),
              row_names_gp = gpar(fontsize = 12, alpha = 0.75, border = FALSE)) #Create a heatmap object
draw(ht) #Draw heatmap
#dev.off()
```

## Plot mean eigengene values for each module

```{r}
#Ensure metadata columns are factors
metabolomicsMetadata$crab.ID <- as.factor(metabolomicsMetadata$crab.ID)
metabolomicsMetadata$treatment <- as.factor(metabolomicsMetadata$treatment)
metabolomicsMetadata$day <- as.factor(metabolomicsMetadata$day)
metabolomicsMetadata$treatment_day <- as.factor(metabolomicsMetadata$treatment_day)
```

```{r}
mME_meta <- merge(mME, metabolomicsMetadata, by = "crab.ID") %>%
  rename(Module = name) #Create new dataframe with metadata
head(mME_meta) #Confirm dataframe creation
mME_meta$treatment_day <- factor(mME_meta$treatment_day, levels = c("5_3", "5_22", 
                                                                    "13_3", "13_22", 
                                                                    "30_3", "30_22")) #Reorder levels
levels(mME_meta$treatment_day) #Confirm levels were reordered
```

```{r}
write.csv(WCNA/mME_meta, "module-expression.csv", quote = FALSE, row.names = FALSE) #Save module expression dataframe
```

First, I'm going to look at boxplots for eigengene expression by treatment x day.

```{r, warning = FALSE, message = FALSE}
#pdf("WCNA/figures/eigengene-expression-boxplot.pdf", height = 8.5, width = 11)
mME_meta %>%
  ggplot(aes(x = treatment_day, y = value, colour = treatment_day)) +
  facet_wrap(~ Module) +
  ylab("Module Expression") +
  geom_hline(yintercept = 0, linetype = "dashed", color = "grey")+
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  stat_summary(fun=mean, geom="line", aes(group = treatment_day, color = treatment_day), position = position_dodge(width = 0.5))  + 
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_text(color = "white"),
                                        axis.ticks.x = element_blank()) #Create a boxplot with data points for module expression on the y-axis and values for each treatment x day combination. Facet by module eigengene. Add a horizontal line at y = 0.
#dev.off()
```

I'm going to make a similar plot, but use dots instead to see if that makes the trends clearer.

```{r}
#pdf("WCNA/figures/eigengene-expression-dotplot.pdf", height = 8.5, width = 11)
mME_meta %>%
  ggplot(aes(x = treatment_day, y = value)) +
  facet_grid(~Module) +
  ylab("Eigengene Expression") +
  geom_hline(yintercept = 0, linetype ="dashed", color = "grey")+
  geom_point(aes(fill = treatment_day, color = treatment_day), size = 3, pch = 21, position = position_dodge(width = 0.5)) +
  geom_smooth(aes(group=1), se=TRUE, show.legend = NA, color="black")+
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 14, color = "black", face="bold"))
#dev.off()
```

```{r}
mME_meta$treatment_day <- factor(mME_meta$treatment_day, levels = c("13_3", "13_22", 
                                                                    "5_3", "5_22", 
                                                                    "30_3", "30_22")) #Reorder levels
levels(mME_meta$treatment_day) #Confirm levels were reordered
```


```{r}
metab_rel_change <- mME_meta %>%
  dplyr::select(treatment_day, crab.ID, Module, value) %>%
  group_by(treatment_day, Module) %>%
  summarise(mean = mean(value)) %>%
  group_by(treatment_day, Module) %>%
  arrange(treatment_day, .by_group = TRUE) %>%
  group_by(Module) %>%
  mutate(metabolite_expression = ((mean-dplyr::first(mean))/abs(dplyr::first(mean)))) %>%
  dplyr::select(Module, treatment_day, metabolite_expression) #Take the dataframe with module information and metadata, and keep only the treatment_day, crab.ID, module, and module eigengene values. Group by treatment_day and Module, then calculate mean eigengene expression. Arrange by treatment_day, group by module again, then calculaterelative change in expression for each module relative to 13ºC at day 3. 
metab_rel_change #Confirm calculations
```

```{r}
write.csv(WCNA/metab_rel_change, "metabolites_relative_change.csv", quote = FALSE, row.names = FALSE)
```

# Create lists of metabolites in each module

For each module, I want the metabolite name, but I also want all of the PubChem and KEGG ID information provided by the sequencing facility.

```{r}
module0_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 0) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module0_metabolites) #40 metabolites
head(module0_metabolites)
write.csv(module0_metabolites, "metaboanalyst/module0_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module1_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 1) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module1_metabolites) #22 metabolites
head(module1_metabolites)
write.csv(module1_metabolites, "metaboanalyst/module1_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module2_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 2) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module2_metabolites) #22 metabolites
head(module2_metabolites)
write.csv(module2_metabolites, "metaboanalyst/module2_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module3_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 3) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module3_metabolites) #19 metabolites
head(module3_metabolites)
write.csv(module3_metabolites, "metaboanalyst/module3_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module4_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 4) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module4_metabolites) #15 metabolites
head(module4_metabolites)
write.csv(module4_metabolites, "metaboanalyst/module4_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module5_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 5) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module5_metabolites) #10 metabolites
head(module5_metabolites)
write.csv(module5_metabolites, "metaboanalyst/module5_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module6_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 6) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module6_metabolites) #10 metabolites
head(module6_metabolites)
write.csv(module6_metabolites, "metaboanalyst/module6_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module7_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 7) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module7_metabolites) #8 metabolites
head(module7_metabolites)
write.csv(module7_metabolites, "metaboanalyst/module7_metabolites.csv", quote = FALSE, row.names = FALSE)
```

```{r}
module8_metabolites <- module_df %>%
  dplyr::rename(., Module = colors) %>%
  filter(., Module == 8) %>%
  dplyr::select(., Metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(Metabolite = BinBase.name)), by = "Metabolite") %>%
  dplyr::select(., Metabolite, PubChem, KEGG) #Take dataframe with modules and metabolites, change colors to Module, and select metabolites for a given module. Keep only the metabolite column. Then join with the raw metabolomics data to get PubChem and KEGG ids.
nrow(module8_metabolites) #6 metabolites
head(module8_metabolites)
write.csv(module8_metabolites, "metaboanalyst/module8_metabolites.csv", quote = FALSE, row.names = FALSE)
```

# Pathway analysis

The final step in this workflow is to understand the various metabolic pathways represented in each module eigengene. To do this, I will conduct a manual pathway analysis and identify VIP present in each significant cluster, and also use MetaboAnalyst.

```{bash}
mkdir metaboanalyst
mkdir metaboanalyst/figures
```

## Manual pathway analysis

There are a few pathways of interest based on my experiment. Since thermal tolerance in marine invertebrates is linked to oxygen availability, the tradeoff between aerobic and anaerobic respiration can tell me if a crab was tipping into a pejus or critical range. Additionally, previous research shows that magnesium can block calcium channels in colder temperatures, reducing mobility. I wanted to see if any metabolites associated with calcium signaling had different abundances according to temperature, which may provide additional evidence for magnesium blockage.

### Aerobic respiration

#### Glycolysis

1. Glucose: Glucose is the starting substrate for glycolysis. It is phosphorylated to form glucose-6-phosphate by hexokinase or glucokinase.
2. Glucose-1-phosphate: Glucose-1-phosphate is an intermediate in the breakdown of glycogen, which can be converted into glucose-6-phosphate before entering glycolysis.
3. Glucose-6-phosphate: It is the first intermediate in glycolysis, formed from glucose by hexokinase. It is subsequently converted into fructose-6-phosphate.
4. Fructose-6-phosphate: It is an intermediate in glycolysis, formed from glucose-6-phosphate by phosphoglucose isomerase. It is subsequently phosphorylated to fructose-1,6-bisphosphate.
5. Fructose-1-phosphate: Fructose-1-phosphate is an intermediate in fructose metabolism. It is converted into glyceraldehyde-3-phosphate (an intermediate in glycolysis) and dihydroxyacetone phosphate by aldolase.
6. 3-Phosphoglycerate: It is an intermediate in glycolysis, formed from 1,3-bisphosphoglycerate by phosphoglycerate kinase. It is subsequently converted into 2-phosphoglycerate.

```{r}
glycolysis_facets <- c("1" = "glucose",
                       "2" = "glucose-1-phosphate",
                       "3" = "glucose-6-phosphate",
                       "4" = "fructose-6-phosphate",
                       "5" = "fructose-1-phosphate",
                       "6" = "3-phosphoglycerate") #Assign labels for facets
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, 
                     `glucose `, `glucose-1-phosphate`, `glucose-6-phosphate`, `fructose-6-phosphate `, `fructose-1-phosphate`, `3-phosphoglycerate`)) %>%
  pivot_longer(., cols = c(5:10), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "glucose " ~ 1,
                              metabolite == "glucose-1-phosphate" ~ 2,
                              metabolite == "glucose-6-phosphate" ~ 3,
                              metabolite == "fructose-6-phosphate " ~ 4,
                              metabolite == "fructose-1-phosphate" ~ 5,
                              metabolite == "3-phosphoglycerate" ~ 6)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = glycolysis_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/glycolysis-plot.pdf", height = 8.5, width = 11)
```

#### TCA cycle

- Citric acid: Also known as citrate, it is the first intermediate in the TCA cycle. It is formed from acetyl-CoA and oxaloacetate and undergoes a series of reactions to regenerate oxaloacetate, producing NADH, FADH2, and GTP in the process.
- Alpha-ketoglutarate: Also known as 2-oxoglutarate, it is an intermediate in the TCA cycle. It is formed from isocitrate by isocitrate dehydrogenase and is subsequently converted into succinyl-CoA.
- Fumaric acid: Fumarate is an intermediate in the TCA cycle, formed by the hydration of fumarate catalyzed by fumarase. It is subsequently converted into malate.
- Malic acid: Malate is an intermediate in the TCA cycle, formed from fumarate by fumarase. It is subsequently converted into oxaloacetate.

```{r}
TCA_facets <- c("1" = "citric acid",
                "2" = "alpha-ketoglutarate",
                "3" = "fumaric acid",
                "4" = "malic acid") #Assign labels for facets
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, 
                     `citric acid`, `alpha-ketoglutarate`, `fumaric acid`, `malic acid`)) %>%
  pivot_longer(., cols = c(5:8), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "citric acid" ~ 1,
                              metabolite == "alpha-ketoglutarate" ~ 2,
                              metabolite == "fumaric acid" ~ 3,
                              metabolite == "malic acid" ~ 4)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = TCA_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/TCA-plot.pdf", height = 8.5, width = 11)
```

### Anaerobic respiration

1. Glucose: Glucose is the starting substrate for glycolysis, the metabolic pathway that generates pyruvate, the precursor of lactic acid.
2. Glucose-1-phosphate: Glucose-1-phosphate is an intermediate in the breakdown of glycogen, which can be converted into glucose-6-phosphate before entering glycolysis.
3. Glucose-6-phosphate: It is the first intermediate in glycolysis, formed from glucose by hexokinase. It is subsequently converted into fructose-6-phosphate.
4. Fructose-6-phosphate: It is an intermediate in glycolysis, formed from glucose-6-phosphate by phosphoglucose isomerase. It is subsequently phosphorylated to fructose-1,6-bisphosphate.
5. Fructose-1-phosphate: Fructose-1-phosphate is an intermediate in fructose metabolism. It is converted into glyceraldehyde-3-phosphate (an intermediate in glycolysis) and dihydroxyacetone phosphate by aldolase.
6. 3-Phosphoglycerate: It is an intermediate in glycolysis, formed from 1,3-bisphosphoglycerate by phosphoglycerate kinase. It is subsequently converted into 2-phosphoglycerate.
7. Lactic acid: The end product of lactic acid fermentation, formed from the reduction of pyruvate.

```{r}
anaerobic_facets <- c(glycolysis_facets,
                      "7" = "lactic acid") #Make facet labels
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     `glucose `, `glucose-1-phosphate`, `glucose-6-phosphate`, `fructose-6-phosphate `, `fructose-1-phosphate`, `3-phosphoglycerate`, `lactic acid`)) %>%
  pivot_longer(., cols = c(5:11), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "glucose " ~ 1,
                              metabolite == "glucose-1-phosphate" ~ 2,
                              metabolite == "glucose-6-phosphate" ~ 3,
                              metabolite == "fructose-6-phosphate " ~ 4,
                              metabolite == "fructose-1-phosphate" ~ 5,
                              metabolite == "3-phosphoglycerate" ~ 6,
                              metabolite == "lactic acid" ~ 7)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free_y", labeller = labeller(order = anaerobic_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/anaerobic-metabolism-plot.pdf", height = 8.5, width = 11)
```

### Combined respiration plot

I'm going to combine metabolite abundance information from aerobic and anaerobic respiration pathways using InDesign. To do this, I need individual plots for each metabolite that are the same size.

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     `glucose `, `glucose-1-phosphate`, `glucose-6-phosphate`, `fructose-6-phosphate `, `fructose-1-phosphate`, `3-phosphoglycerate`,
                     `lactic acid`,
                     `citric acid`, `alpha-ketoglutarate`, `fumaric acid`, `malic acid`)) %>%
  pivot_longer(., cols = c(5:15), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~metabolite, scales = "free") +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/combined-respiration-inputs.pdf", height = 8.5, width = 11)
```

```{r}
respirationMetabolites <- c("glucose ", "glucose-1-phosphate", "glucose-6-phosphate", "fructose-6-phosphate ", "fructose-1-phosphate", "3-phosphoglycerate",
                            "lactic acid",
                            "citric acid", "alpha-ketoglutarate", "fumaric acid", "malic acid") #Make a vector with metabolites of interest
```

```{r}
overallVIP_respirationMetabolites <- significantVIP %>%
  filter(., metabolite %in% respirationMetabolites) #Identify overall VIP associated with pathway
overallVIP_respirationMetabolites
```

```{r}
write.table(overallVIP_respirationMetabolites, "metaboanalyst/combined-respiration-overall-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
VIP_respirationMetabolites <- rbind(t.test_13v30_day3 %>%
                                      mutate(., comparison = "13v30_day3") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13v5_day3 %>%
                                      mutate(., comparison = "13v5_day3") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13v30_day22 %>%
                                      mutate(., comparison = "13v30_day22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13v5_day22 %>%
                                      mutate(., comparison = "13v5_day22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_5_day3v22 %>%
                                      mutate(., comparison = "5_day3v22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_13_day3v22 %>%
                                      mutate(., comparison = "13_day3v22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites),
                                    t.test_30_day3v22 %>%
                                      mutate(., comparison = "30_day3v22") %>%
                                      filter(., p.adj < 0.05) %>%
                                      filter(., metabolite %in% respirationMetabolites))
VIP_respirationMetabolites
```

```{r}
write.table(VIP_respirationMetabolites, "metaboanalyst/combined-respiration-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

### Calcium signaling

- UDP-N-acetylglucosamine: Calcium signaling can influence the activity of enzymes involved in the synthesis of UDP-N-acetylglucosamine. Blocking calcium channels with Mg2+ could potentially reduce the production of UDP-N-acetylglucosamine, leading to decreased concentrations.
- Tyrosine: Calcium signaling pathways can regulate the activity of enzymes involved in tyrosine metabolism. Blocking calcium channels may disrupt these pathways, potentially altering the concentration of tyrosine.
- Tryptophan: Similar to tyrosine, tryptophan metabolism can be influenced by calcium signaling. Blocking calcium channels may affect tryptophan metabolism and alter its concentration.
- Threonine: Threonine is an amino acid involved in various metabolic pathways. Changes in calcium signaling can affect threonine metabolism, potentially leading to alterations in its concentration.
Serotonin: Calcium signaling pathways play a crucial role in the release and regulation of serotonin in smooth muscle cells. Blocking calcium channels with Mg2+ could lead to decreased serotonin release and lower concentrations.
- Serine: Similar to threonine, serine is an amino acid involved in various metabolic pathways. Changes in calcium signaling can affect serine metabolism, potentially leading to alterations in its concentration.
- Phosphoinositol: Phosphoinositol lipids are components of cell membranes and can serve as precursors for signaling molecules such as inositol trisphosphate (IP3), which plays a key role in intracellular calcium release from the endoplasmic reticulum.
- Citric acid: Citric acid is an intermediate in the citric acid cycle, which can be influenced by calcium signaling pathways. Blocking calcium channels may disrupt the citric acid cycle, potentially leading to alterations in citric acid concentration.
- Glutathione: Calcium signaling pathways can regulate the synthesis and metabolism of glutathione. Blocking calcium channels may disrupt glutathione metabolism, potentially altering its concentration.
- Glutamine: Glutamine is an amino acid involved in various metabolic pathways. Changes in calcium signaling can affect glutamine metabolism, potentially leading to alterations in its concentration.
- Glutamate: Glutamate is an excitatory neurotransmitter that can be influenced by calcium signaling pathways. Blocking calcium channels may disrupt glutamate release and metabolism, potentially altering its concentration.

```{r}
calciumSignalingMetabolites <- c("UDP-N-acetylglucosamine ", "tyrosine", "tryptophan ", "threonine", "serine", "phosphoinositol", "citric acid", "glutathione ", "glutamine", "glutamate ") #Create a vector with metabolites of interest
```

```{r}
overallVIP_calciumSignalingMetabolites <- significantVIP %>%
  filter(., metabolite %in% calciumSignalingMetabolites) #Identify overall VIP associated with pathway
overallVIP_calciumSignalingMetabolites
```

```{r}
write.table(overallVIP_calciumSignalingMetabolites, "metaboanalyst/calcium-signaling-overall-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
VIP_calciumSignalingMetabolites <- rbind(t.test_13v30_day3 %>%
                                           mutate(., comparison = "13v30_day3") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13v5_day3 %>%
                                           mutate(., comparison = "13v5_day3") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13v30_day22 %>%
                                           mutate(., comparison = "13v30_day22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13v5_day22 %>%
                                           mutate(., comparison = "13v5_day22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_5_day3v22 %>%
                                           mutate(., comparison = "5_day3v22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_13_day3v22 %>%
                                           mutate(., comparison = "13_day3v22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites),
                                         t.test_30_day3v22 %>%
                                           mutate(., comparison = "30_day3v22") %>%
                                           filter(., p.adj < 0.05) %>%
                                           filter(., metabolite %in% calciumSignalingMetabolites))
VIP_calciumSignalingMetabolites
```

```{r}
write.table(VIP_calciumSignalingMetabolites, "metaboanalyst/calcium-signaling-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
calciumSignaling_facets <- c("1" = "glutamine",
                             "2" = "glutathione ",
                             "3" = "threonine",
                             "4" = "tryptophan ",
                             "5" = "UDP-N-acetylglucosamine ",
                             "6" = "serine",
                             "7" = "citric acid",
                             "8" = "glutamate ",
                             "9" = "phosphoinositol",  
                             "10" = "tyrosine") #Create facet labels
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     `UDP-N-acetylglucosamine `, tyrosine, `tryptophan `, threonine, serine, phosphoinositol, `citric acid`, `glutathione `, glutamine, `glutamate `)) %>%
  pivot_longer(., cols = c(5:14), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "glutamine" ~ 1,
                              metabolite == "glutathione " ~ 2,
                              metabolite == "threonine" ~ 3,
                              metabolite == "tryptophan " ~ 4,
                              metabolite == "UDP-N-acetylglucosamine " ~ 5,
                              metabolite == "serine" ~ 6,
                              metabolite == "citric acid" ~7,
                              metabolite == "glutamate " ~ 8,
                              metabolite == "phosphoinositol" ~ 9,
                              metabolite == "tyrosine" ~ 10)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free", labeller = labeller(order = calciumSignaling_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/calcium-signaling-plot.pdf", height = 8.5, width = 11)
```

### Cold protectors

```{r}
coldProtectionMetabolites <- c("alanine", "ornithine ", "proline") #Create a vector with metabolites of interest
```

```{r}
overallVIP_coldProtectionMetabolites <- significantVIP %>%
  filter(., metabolite %in% coldProtectionMetabolites) #Identify overall VIP associated with pathway
overallVIP_coldProtectionMetabolites
```

```{r}
VIP_coldProtectionMetabolites <- rbind(t.test_13v30_day3 %>%
                                         mutate(., comparison = "13v30_day3") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13v5_day3 %>%
                                         mutate(., comparison = "13v5_day3") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13v30_day22 %>%
                                         mutate(., comparison = "13v30_day22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13v5_day22 %>%
                                         mutate(., comparison = "13v5_day22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_5_day3v22 %>%
                                         mutate(., comparison = "5_day3v22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_13_day3v22 %>%
                                         mutate(., comparison = "13_day3v22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites),
                                       t.test_30_day3v22 %>%
                                         mutate(., comparison = "30_day3v22") %>%
                                         filter(., p.adj < 0.05) %>%
                                         filter(., metabolite %in% coldProtectionMetabolites))
VIP_coldProtectionMetabolites
```

```{r}
write.table(VIP_coldProtectionMetabolites, "metaboanalyst/cold-protection-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

```{r}
coldProtection_facets <- c("1" = "ornithine ",
                           "2" = "proline",
                           "3" = "alanine") #Create facet labels
```

```{r}
transMetabData %>%
  dplyr::select(., c(1:4,
                     alanine, `ornithine `, proline)) %>%
  pivot_longer(., cols = c(5:7), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  mutate(., order = case_when(metabolite == "ornithine " ~ 1,
                              metabolite == "proline" ~ 2,
                              metabolite == "alanine" ~ 3)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~order, scales = "free", labeller = labeller(order = coldProtection_facets)) +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/cold-protection-plot.pdf", height = 8.5, width = 11)
```

## VIP "enrichment" analysis

The second step of this workflow is to analyze the abundance patterns and associated functions/pathways of significant VIP.

### All VIP

#### Make heatmaps

```{r}
heatmapColors <- rev(brewer.pal("PRGn", n = 9))
```

```{r}
metabolomicsAnnotation <- metabolomicsMetadata %>%
  mutate(., Treatment = case_when(treatment_day == "13_3" ~ "13ºC at Day 3",
                                  treatment_day == "13_22" ~ "13ºC at Day 22",
                                  treatment_day == "5_3" ~ "5ºC at Day 3",
                                  treatment_day == "5_22" ~ "5ºC at Day 22",
                                  treatment_day == "30_3" ~ "30ºC at Day 3",
                                  treatment_day == "30_22" ~ "30ºC at Day 22"))
rownames(metabolomicsAnnotation) <- metabolomicsAnnotation$crab.ID #Change crab ID to rownames
metabolomicsAnnotation <- metabolomicsAnnotation %>%
  dplyr::select(Treatment) #Remove crab ID column
head(metabolomicsAnnotation) #Confirm changes
```

```{r}
metabolomicsAvgAnnotation <- metabolomicsMetadata %>%
  mutate(., Treatment = case_when(treatment_day == "13_3" ~ "13ºC at Day 3",
                                  treatment_day == "13_22" ~ "13ºC at Day 22",
                                  treatment_day == "5_3" ~ "5ºC at Day 3",
                                  treatment_day == "5_22" ~ "5ºC at Day 22",
                                  treatment_day == "30_3" ~ "30ºC at Day 3",
                                  treatment_day == "30_22" ~ "30ºC at Day 22")) %>%
  dplyr::select(treatment_day, Treatment) %>%
  unique(.) #Create treatment column. Keep only columns of interest and retain unique rows.
rownames(metabolomicsAvgAnnotation) <- metabolomicsAvgAnnotation$treatment_day #Change treatment_day to rownames
metabolomicsAvgAnnotation <- metabolomicsAvgAnnotation %>%
  dplyr::select(-treatment_day)
head(metabolomicsAvgAnnotation) #Confirm changes
```

```{r}
metabolomicsColors <- list(Treatment = c("13ºC at Day 3" = "grey80",
                                         "13ºC at Day 22" = "#525252",
                                         "5ºC at Day 3" = "lightblue",
                                         "5ºC at Day 22" = "#2171B5",
                                         "30ºC at Day 3" = "salmon",
                                         "30ºC at Day 22" = "#CB181D")) #List of colors for column annotations
metabolomicsColors #Column annotation colors
```

```{r}
transMetabData$treatment_day <- factor(transMetabData$treatment_day, 
                                       levels=c("13_3",
                                                "13_22", 
                                                "5_3",
                                                "5_22",
                                                "30_3",
                                                "30_22")) #Reorder levels
```

```{r}
#pdf("metaboanalyst/figures/all_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(crab.ID, any_of(significantVIP$metabolite)) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors)
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/all_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/all_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

```{r}
#svg("metaboanalyst/figures/all_VIP-zscore.svg", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

#### MetaboAnalyst 

The next thing I want to do with the list of all VIP is to add PubChem and KEGG annotations. This will allow me to upload the list into MetaboAnalyst to conduct pathway assignment and check for enriched pathways.

```{r}
significantVIP %>%
  dplyr::select(., metabolite) %>% 
  left_join(., (rawMetabolomicsData %>% dplyr::rename(metabolite = BinBase.name)), by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  write.table("metaboanalyst/allVIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take list of significant VIP. Select only the metabolites and join with annotation information. Select only columns of interest and save as a txt file
```

Within MetaboAnalyst, I did the following:

- Uploaded list of metabolite compound names for significant VIP in the Pathway Analysis module
- Matched to KEGG pathways from *Strongylocentrotus purpuratus*

Since the Pathway Analysis module integrates "enrichment analysis and pathway topology analysis," I did not conduct any other analyses. Now I will create figures based on the MetaboAnalyst Pathway Analysis output.

```{r}
pathwayResults <- read.csv("metaboanalyst/allVIP_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected)
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/all_VIP-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
topPathwayResults <- pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(topPathwayResults) #Confirm formatting
```

```{r}
topPathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  xlim(0,16) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/all_VIP-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
topPathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  xlim(0,16) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/all_VIP-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
topPathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/all_VIP-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

I want to make a list of significant VIP involved in significant pathways. I can then see if any of these pathways are represented in the modules. I matched names of VIP from MetaboAnalyst using the names and KEGG IDs in the name map file.

```{r}
#Create lists of metabolites involved in significant pathways. Some names have spaces afterwards to reflect naming conventions in original dataframe
valineBiosynthesis <- c("threonine", "isoleucine ", "leucine", "2-ketoisocaproic acid", "valine ")
arginineBiosynthesis <- c("alpha-ketoglutarate", "glutamate ", "citrulline", "ornithine ", "urea", "fumaric acid")
glutathioneMetabolism <- c("glycine", "oxoproline", "glutamate ", "ornithine ", "putrescine", "cadaverine")
alanineMetabolism <- c("asparagine", "fumaric acid", "alanine", "alpha-ketoglutarate", "glutamate ")
```

```{r}
significantVIP_significantPathways <- rbind(significantVIP %>%
                                              filter(., metabolite %in% valineBiosynthesis) %>%
                                              mutate(., pathway = "valineBiosynthesis"),
                                            significantVIP %>%
                                              filter(., metabolite %in% arginineBiosynthesis) %>%
                                              mutate(., pathway = "arginineBiosynthesis"),
                                            significantVIP %>%
                                              filter(., metabolite %in% glutathioneMetabolism) %>%
                                              mutate(., pathway = "glutathioneMetabolism"),
                                            significantVIP %>%
                                              filter(., metabolite %in% alanineMetabolism) %>%
                                              mutate(., pathway = "alanineMetabolism"))
significantVIP_significantPathways
```

```{r}
write.table(significantVIP_significantPathways, "metaboanalyst/significant-VIP-significant-pathways.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

##### Valine, leucine, and isoleucine biosynthesis

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, any_of(valineBiosynthesis))) %>%
  pivot_longer(., cols = c(5:9), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~metabolite, scales = "free_y") +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/valine-biosynthesis-plot.pdf", height = 8.5, width = 11)
```

##### Arginine biosynthesis

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, any_of(arginineBiosynthesis))) %>%
  pivot_longer(., cols = c(5:10), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~metabolite, scales = "free_y") +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/arginine-biosynthesis-plot.pdf", height = 8.5, width = 11)
```

##### Glutathione metabolism

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, any_of(glutathioneMetabolism))) %>%
  pivot_longer(., cols = c(5:10), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~metabolite, scales = "free_y") +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/glutathione-metabolism-plot.pdf", height = 8.5, width = 11)
```

##### Alanine, aspartate, and glutamate metabolism

```{r}
transMetabData %>%
  dplyr::select(., c(1:4, any_of(alanineMetabolism))) %>%
  pivot_longer(., cols = c(5:9), names_to = "metabolite", values_to = "value") %>%
  mutate(., treatment_day = gsub(pattern = "_3", replacement = "_03", x = treatment_day)) %>%
  mutate(., treatment_day = gsub(pattern = "5_", replacement = "05_", x = treatment_day)) %>%
  ggplot(., mapping = aes(x = treatment_day, y = value, color = treatment_day)) +
  facet_wrap(~metabolite, scales = "free_y") +
  ylab("Metabolite Abundance") +
  geom_boxplot(width = 0.5, position = position_dodge(width = 0.5), alpha = 0.7) +
  geom_point(pch = 21, position = position_dodge(width = 0.5)) +
  scale_fill_manual(name = "Temperature (ºC) x Day", 
                    values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                    labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  scale_colour_manual(name = "Temperature (ºC) x Day", 
                      values = c("lightblue", "#2171B5", "grey80", "#525252", "salmon", "#CB181D"),
                      labels = c("5ºC on Day 3", "5ºC on Day 22", "13ºC on Day 3", "13ºC on Day 22", "30ºC on Day 3", "30ºC on Day 22")) +
  xlab("") + #Axis titles
  theme_classic(base_size = 15) + theme(axis.text.x = element_blank(), 
                                        axis.ticks.x = element_blank(),
                                        legend.position = "bottom", 
                                        legend.text = element_text(color = "black", size = 12), 
                                        strip.text.x = element_text(size = 10, color = "black", face="bold"))
ggsave("metaboanalyst/figures/alanine-metabolism-plot.pdf", height = 8.5, width = 11)
```

#### Identify VIP in each module

I will only identify VIP in modules that were significantly correlated with a temperature x day condition.

```{r}
overallVIP_module0 <- significantVIP %>%
  filter(., metabolite %in% module0_metabolites$Metabolite) %>%
  mutate(module = "module0") #Identify overall VIP present in module
overallVIP_module0
```

```{r}
overallVIP_module2 <- significantVIP %>%
  filter(., metabolite %in% module2_metabolites$Metabolite) %>%
  mutate(module = "module2") #Identify overall VIP present in module
overallVIP_module2
```

```{r}
overallVIP_module3 <- significantVIP %>%
  filter(., metabolite %in% module3_metabolites$Metabolite) %>%
  mutate(module = "module3") #Identify overall VIP present in module
overallVIP_module3
```

```{r}
overallVIP_module4 <- significantVIP %>%
  filter(., metabolite %in% module4_metabolites$Metabolite) %>%
  mutate(module = "module4") #Identify overall VIP present in module
overallVIP_module4
```

```{r}
overallVIP_module5 <- significantVIP %>%
  filter(., metabolite %in% module5_metabolites$Metabolite) %>%
  mutate(module = "module5") #Identify overall VIP present in module
overallVIP_module5
```

```{r}
overallVIP_module6 <- significantVIP %>%
  filter(., metabolite %in% module6_metabolites$Metabolite) %>%
  mutate(module = "module6") #Identify overall VIP present in module
overallVIP_module6
```

```{r}
overallVIP_module7 <- significantVIP %>%
  filter(., metabolite %in% module7_metabolites$Metabolite) %>%
  mutate(module = "module7") #Identify overall VIP present in module
overallVIP_module7
```

```{r}
overallVIP_allModules <- rbind(overallVIP_module0,
                               overallVIP_module2,
                               overallVIP_module3,
                               overallVIP_module4,
                               overallVIP_module5,
                               overallVIP_module6,
                               overallVIP_module7)
overallVIP_allModules
```

```{r}
write.table(overallVIP_allModules, "metaboanalyst/all-module-overall-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Identify VIP from significant pathways in each module

```{r}
module0_VIPpathway <- rbind(module0_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module0_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module0_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module0_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module0") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module0_VIPpathway
```

```{r}
module2_VIPpathway <- rbind(module2_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module2_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module2_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module2_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module2") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module2_VIPpathway
```

```{r}
module3_VIPpathway <- rbind(module3_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module3_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module3_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module3_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module3") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module3_VIPpathway
```

```{r}
module4_VIPpathway <- rbind(module4_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module4_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module4_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module4_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module4") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module4_VIPpathway
```

```{r}
module5_VIPpathway <- rbind(module5_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module5_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module5_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module5_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module5") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module5_VIPpathway
```

```{r}
module6_VIPpathway <- rbind(module6_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module6_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module6_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module6_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module6") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module6_VIPpathway
```

```{r}
module7_VIPpathway <- rbind(module7_metabolites %>%
                              filter(., Metabolite %in% valineBiosynthesis) %>%
                              mutate(., pathway = "valineBiosynthesis"),
                            module7_metabolites %>%
                              filter(., Metabolite %in% alanineMetabolism) %>%
                              mutate(., pathway = "alanineMetabolism"),
                            module7_metabolites %>%
                              filter(., Metabolite %in% glutathioneMetabolism) %>%
                              mutate(., pathway = "glutathioneMetabolism"),
                            module7_metabolites %>%
                              filter(., Metabolite %in% arginineBiosynthesis) %>%
                              mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module7") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module7_VIPpathway
```

```{r}
VIPpathway_allModules <- rbind(module0_VIPpathway,
                               module2_VIPpathway,
                               module3_VIPpathway, 
                               module4_VIPpathway,
                               module5_VIPpathway,
                               module6_VIPpathway,
                               module7_VIPpathway) #Combine dataframes
```

```{r}
write.table(VIPpathway_allModules, "metaboanalyst/all-module-VIP-pathways.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

### Pairwise VIP

#### MetaboAnalyst

I'll conduct enrichment tests using pairwise VIP from my various contrasts, since that will shed more light on short- and long-term acclimation processes to various temperatures.

```{r}
t.test_30v5_day3 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/30v5-day3_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_13v30_day22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/13v30-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_13v5_day22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/13v5-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_30v5_day22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("../output/05-metabolomics-analysis/metaboanalyst/30v5-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_30v5_day22 %>%
  filter(., p.adj < 0.05) %>%
  anti_join(x = ., y = (t.test_13v5_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  anti_join(x = ., y = (t.test_13v30_day22 %>%
                          ungroup(.) %>%
                          filter(., p.adj < 0.05)),
            by = "metabolite") %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("../output/05-metabolomics-analysis/metaboanalyst/unique-30v5-day22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

```{r}
t.test_5_day3v22 %>%
  filter(., p.adj < 0.05) %>%
  left_join(., (rawMetabolomicsData %>% 
                  dplyr::rename(metabolite = BinBase.name)), 
            by = "metabolite") %>%
  dplyr::select(., metabolite, PubChem, KEGG) %>%
  unique(.) %>%
  write.table("metaboanalyst/5-day3v22_VIP_metabolites.txt", sep = "\t", quote = FALSE, row.names = FALSE) #Take significant pairwise VIP data and join with lipid identifiers. Retain unique entries and write table
```

Within MetaboAnalyst, I did the following:

- Uploaded each list of metabolite compound names for significant VIP in the Pathway Analysis module
- Matched to KEGG pathways from *Drosophila melanogaster*

##### Day 3: 30ºC vs. 5ºC

```{r}
day3_30v5_pathwayResults <- read.csv("metaboanalyst/30v5-day3_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day3_30v5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day3_30v5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day3-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day3_30v5_pathwayResults <- day3_30v5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day3_30v5_pathwayResults) #Confirm formatting
```

```{r}
top_day3_30v5_pathwayResults %>%
  mutate(., order = rev(1:12)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_day3_30v5_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day3-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

##### Day 22: 13ºC vs. 30ºC

```{r}
significantVIP_13v30_day22 <- t.test_13v30_day22 %>%
  filter(., p.adj < 0.05) #Create an object with significant VIP from the specific comparison
head(significantVIP_13v30_day22) #Confirm dataframe creation
nrow(significantVIP_13v30_day22)
```

```{r}
#pdf("metaboanalyst/figures/13v30_day22_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
            group_by(treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% as.data.frame(.) %>%
            log(.) %>%
            replace(is.na(.), 0) %>%
            as.matrix()), cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "30ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/13v30_day22_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
            group_by(treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            replace(is.na(.), 0) %>%
            mutate(across(where(is.numeric), scale)) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), 
         cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "30ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Z-Score")
#dev.off()
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `30_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "30")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[1]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v30-day22-bargraph.pdf", height = 8.5, width = 11)
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v30_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "30_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `30_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "30")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[1]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v30-day22-bargraph.svg", height = 8.5, width = 11)
```

```{r}
day22_13v30_pathwayResults <- read.csv("metaboanalyst/13v30-day22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day22_13v30_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day22_13v30_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v30_pathwayResults <- day22_13v30_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day22_13v30_pathwayResults) #Confirm formatting
```

```{r}
top_day22_13v30_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v30_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v30_pathwayResults %>%
  mutate(., order = rev(1:25)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v30_day22-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

##### Day 22: 13ºC vs. 5ºC

```{r}
significantVIP_13v5_day22 <- t.test_13v5_day22 %>%
  filter(., p.adj < 0.05) #Create an object with significant VIP from the specific comparison
head(significantVIP_13v5_day22) #Confirm dataframe creation
nrow(significantVIP_13v5_day22)
```

```{r}
#pdf("metaboanalyst/figures/13v5_day22_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% as.data.frame(.) %>%
            log(.) %>%
            replace(is.na(.), 0) %>%
            as.matrix()), cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/13v5_day22_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
            filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            replace(is.na(.), 0) %>%
            mutate(across(where(is.numeric), scale)) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), 
         cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "13ºC at Day 22" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Z-Score")
#dev.off()
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "5")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[3]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v5-day22-bargraph.pdf", height = 8.5, width = 11)
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_13v5_day22$metabolite)) %>%
  filter(treatment_day == "13_22" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `13_22` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "13",
                               logDiff < 0 ~ "5")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c(plotColors[2], plotColors[3]), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-13v5-day22-bargraph.svg", height = 8.5, width = 11)
```

```{r}
day22_13v5_pathwayResults <- read.csv("metaboanalyst/13v5-day22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day22_13v5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day22_13v5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 2.5) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v5_pathwayResults <- day22_13v5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day22_13v5_pathwayResults) #Confirm formatting
```

```{r}
top_day22_13v5_pathwayResults %>%
  mutate(., order = rev(1:23)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v5_pathwayResults %>%
  mutate(., order = rev(1:23)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
top_day22_13v5_pathwayResults %>%
  mutate(., order = rev(1:23)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/13v5_day22-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

##### Day 22: 30ºC vs. 5ºC

```{r}
day22_30v5_pathwayResults <- read.csv("metaboanalyst/unique-30v5-day22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day22_30v5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day22_30v5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 6) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day22_30v5_pathwayResults <- day22_30v5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day22_30v5_pathwayResults) #Confirm formatting
```

```{r}
top_day22_30v5_pathwayResults %>%
  mutate(., order = rev(1:10)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(top_day22_30v5_pathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/30v5_day22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

##### 5ºC: Day 3 vs. Day 22

```{r}
significantVIP_5_day3v22 <- t.test_5_day3v22 %>%
  filter(., p.adj < 0.05) #Create an object with significant VIP from the specific comparison
head(significantVIP_5_day3v22) #Confirm dataframe creation
nrow(significantVIP_5_day3v22)
```

```{r}
#pdf("metaboanalyst/figures/5_day3v22_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
            filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% as.data.frame(.) %>%
            log(.) %>%
            replace(is.na(.), 0) %>%
            as.matrix()), cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "5ºC at Day 3" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/5_day3v22_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
            filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
            group_by(treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            replace(is.na(.), 0) %>%
            mutate(across(where(is.numeric), scale)) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), 
         cluster_row = TRUE, fontsize_row = 4, show_rownames = TRUE,
         cluster_cols = FALSE, show_colnames = FALSE, 
         color = heatmapColors, 
         annotation_col = metabolomicsAvgAnnotation %>% filter(Treatment == "5ºC at Day 3" | Treatment == "5ºC at Day 22"), 
         annotation_colors = metabolomicsColors, 
         name = "Z-Score")
#dev.off()
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
  filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `5_3` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "3",
                               logDiff < 0 ~ "22")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c("black", "purple"), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-5-day3v22-bargraph.pdf", height = 8.5, width = 11)
```

```{r}
transMetabData %>%
  dplyr::select(treatment_day, any_of(significantVIP_5_day3v22$metabolite)) %>%
  filter(treatment_day == "5_3" | treatment_day == "5_22") %>%
  group_by(treatment_day) %>%
  summarize_all(funs(mean(., na.rm = TRUE))) %>%
  column_to_rownames(var = "treatment_day") %>%
  t(.) %>% as.data.frame(.) %>%
  rownames_to_column(var = "metabolite") %>%
  mutate(., diff = `5_3` - `5_22`) %>%
  mutate(., logDiff = log(abs(diff))) %>%
  mutate(., logDiff = case_when(diff < 0 ~ -logDiff,
                                diff > 0 ~ logDiff)) %>%
  mutate(., status = case_when(logDiff > 0 ~ "3",
                               logDiff < 0 ~ "22")) %>%
  ggplot(., aes(x = logDiff, y = metabolite, fill = status)) +
  geom_bar(aes(fill = status), stat = 'identity') +
  labs(x = "log(Difference)", y = "Metabolite") +
  scale_fill_manual(values = c("black", "purple"), 
                    guide = "none") +
  theme_classic(base_size = 15) + theme(axis.text.y = element_text(size = 6)) #Average treatments of interest for data and transpose. Calculate log difference, negative if higher in the treatment. Create a barplot
ggsave("metaboanalyst/figures/pairwise-VIP-5-day3v22-bargraph.svg", height = 8.5, width = 11)
```

```{r}
day3v22_5_pathwayResults <- read.csv("metaboanalyst/5-day3v22_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR))
#Import pathway assignment data. Rename metabolite column and create new columns (pop observed hits: number of hits/total metabolites in pathway; enrichment ratio: hits/expected hits)
head(day3v22_5_pathwayResults) #Look at pathway analysis results. P-values are from enrichment analysis, impact is from pathway topology analysis
```

```{r}
day3v22_5_pathwayResults %>%
  ggplot(., aes(x = Impact, y = X.log10.p., size = enrichmentRatio, color = Significant)) +
  geom_point(alpha = 0.6) +
  labs(x = "Pathway Impact", y = "Scaled Enrichment P-Value") +
  ylim(0, 3) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-pathwaySignificance.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day3v22_5_pathwayResults <- day3v22_5_pathwayResults %>%
  arrange(desc(enrichmentRatio)) %>%
  slice(., 1:25) #Arrange results by enrichment ratio and take the top 25 metabolites
head(top_day3v22_5_pathwayResults) #Confirm formatting
```

```{r}
top_day3v22_5_pathwayResults %>%
  mutate(., order = rev(1:8)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-enrichmentRatio.pdf", heigh = 8.5, width = 11)
```

```{r}
top_day3v22_5_pathwayResults %>%
  mutate(., order = rev(1:8)) %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

```{r}
top_day3v22_5_pathwayResults %>%
  mutate(., order = rev(1:8)) %>%
  ggplot(., aes(x = X.log10.p., y = as.factor(order), color = Significant, size = enrichmentRatio)) +
  geom_point() +
  labs(x = "-log10(p)") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResults$metabolite)) +
  scale_size_continuous(name = "Enrichment Ratio") +
  scale_color_manual(name = "FDR",
                     values = c("grey20", "springgreen4"),
                     labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("metaboanalyst/figures/5_day3v22-enrichmentOverview.pdf", heigh = 8.5, width = 11)
```

#### Identify pairwise VIP in each module

I'm only interested in the following modules significantly correlated with specific treatment conditions:

- Module 0: 5 and 30 at day 22
- Module 2: 5 at day 3 and 30 at day 22
- Module 3: 5 and 30 at day 22
- Module 4: 30 at a day 22
- Module 5: 5 at day 22, 30 at day 22, 30 at day 3
- Module 6: 13 at day 22
- Module 7: 13 and 5 at day 22

```{r}
VIP_module0 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module0_metabolites$Metabolite)) %>% 
  mutate(module = "module0") #Identify VIP present in module 0 across all pairwise comparisons
VIP_module0
```

```{r}
VIP_module2 <- rbind(t.test_13v5_day3 %>%
                       mutate(., comparison = "13v5_day3") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite),
                     t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module2_metabolites$Metabolite))  %>% 
  mutate(module = "module2") #Identify VIP present in module 2 across all pairwise comparisons
VIP_module2
```

```{r}
VIP_module3 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module3_metabolites$Metabolite))  %>% 
  mutate(module = "module3") #Identify VIP present in module 3 across all pairwise comparisons
VIP_module3
```

```{r}
VIP_module4 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module4_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module4_metabolites$Metabolite))  %>% 
  mutate(module = "module4") #Identify VIP present in module 4 across all pairwise comparisons
VIP_module4
```

```{r}
VIP_module5 <- rbind(t.test_13v30_day3 %>%
                       mutate(., comparison = "13v30_day3") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite),
                     t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite),
                     t.test_30_day3v22 %>%
                       mutate(., comparison = "30_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module5_metabolites$Metabolite))  %>% 
  mutate(module = "module5") #Identify VIP present in module 5 across all pairwise comparisons
VIP_module5
```

```{r}
VIP_module6 <- rbind(t.test_13v30_day22 %>%
                       mutate(., comparison = "13v30_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module6_metabolites$Metabolite),
                     t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module6_metabolites$Metabolite),
                     t.test_13_day3v22 %>%
                       mutate(., comparison = "13_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module6_metabolites$Metabolite))  %>% 
  mutate(module = "module6") #Identify VIP present in module 0 across all pairwise comparisons
VIP_module6
```

```{r}
VIP_module7 <- rbind(t.test_13v5_day22 %>%
                       mutate(., comparison = "13v5_day22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module7_metabolites$Metabolite),
                     t.test_5_day3v22 %>%
                       mutate(., comparison = "5_day3v22") %>%
                       filter(., p.adj < 0.05) %>%
                       filter(., metabolite %in% module7_metabolites$Metabolite))  %>% 
  mutate(module = "module7") #Identify VIP present in module 0 across all pairwise comparisons
VIP_module7
```

```{r}
VIP_allModules <- rbind(VIP_module0,
                        VIP_module2,
                        VIP_module3,
                        VIP_module4,
                        VIP_module5,
                        VIP_module6,
                        VIP_module7) #Combine all VIP lists
head(VIP_allModules)
```

```{r}
write.table(VIP_allModules, "metaboanalyst/all-module-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Identify pairwise VIP that overlap with significant pathways

```{r}
VIP_valineBiosynthesis <- rbind(t.test_13v30_day3 %>%
                                  mutate(., comparison = "13v30_day3") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13v5_day3 %>%
                                  mutate(., comparison = "13v5_day3") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13v30_day22 %>%
                                  mutate(., comparison = "13v30_day22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13v5_day22 %>%
                                  mutate(., comparison = "13v5_day22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_5_day3v22 %>%
                                  mutate(., comparison = "5_day3v22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_13_day3v22 %>%
                                  mutate(., comparison = "13_day3v22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis),
                                t.test_30_day3v22 %>%
                                  mutate(., comparison = "30_day3v22") %>%
                                  filter(., p.adj < 0.05) %>%
                                  filter(., metabolite %in% valineBiosynthesis)) %>%
  mutate(., pathway = "valineBiosynthesis")
VIP_valineBiosynthesis
```

```{r}
VIP_arginineBiosynthesis <- rbind(t.test_13v30_day3 %>%
                                    mutate(., comparison = "13v30_day3") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13v5_day3 %>%
                                    mutate(., comparison = "13v5_day3") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13v30_day22 %>%
                                    mutate(., comparison = "13v30_day22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13v5_day22 %>%
                                    mutate(., comparison = "13v5_day22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_5_day3v22 %>%
                                    mutate(., comparison = "5_day3v22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_13_day3v22 %>%
                                    mutate(., comparison = "13_day3v22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis),
                                  t.test_30_day3v22 %>%
                                    mutate(., comparison = "30_day3v22") %>%
                                    filter(., p.adj < 0.05) %>%
                                    filter(., metabolite %in% arginineBiosynthesis)) %>%
  mutate(., pathway = "arginineBiosynthesis")
VIP_arginineBiosynthesis
```
```{r}
VIP_glutathioneMetabolism <- rbind(t.test_13v30_day3 %>%
                                     mutate(., comparison = "13v30_day3") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13v5_day3 %>%
                                     mutate(., comparison = "13v5_day3") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13v30_day22 %>%
                                     mutate(., comparison = "13v30_day22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13v5_day22 %>%
                                     mutate(., comparison = "13v5_day22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_5_day3v22 %>%
                                     mutate(., comparison = "5_day3v22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_13_day3v22 %>%
                                     mutate(., comparison = "13_day3v22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism),
                                   t.test_30_day3v22 %>%
                                     mutate(., comparison = "30_day3v22") %>%
                                     filter(., p.adj < 0.05) %>%
                                     filter(., metabolite %in% glutathioneMetabolism)) %>%
  mutate(., pathway = "glutathioneMetabolism")
VIP_glutathioneMetabolism
```
```{r}
VIP_alanineMetabolism <- rbind(t.test_13v30_day3 %>%
                                 mutate(., comparison = "13v30_day3") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13v5_day3 %>%
                                 mutate(., comparison = "13v5_day3") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13v30_day22 %>%
                                 mutate(., comparison = "13v30_day22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13v5_day22 %>%
                                 mutate(., comparison = "13v5_day22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_5_day3v22 %>%
                                 mutate(., comparison = "5_day3v22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_13_day3v22 %>%
                                 mutate(., comparison = "13_day3v22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism),
                               t.test_30_day3v22 %>%
                                 mutate(., comparison = "30_day3v22") %>%
                                 filter(., p.adj < 0.05) %>%
                                 filter(., metabolite %in% alanineMetabolism)) %>%
  mutate(., pathway = "alanineMetabolism")
VIP_alanineMetabolism
```

```{r}
VIP_allPathways <- rbind(VIP_valineBiosynthesis,
                         VIP_arginineBiosynthesis,
                         VIP_glutathioneMetabolism,
                         VIP_alanineMetabolism)
VIP_allPathways
```

```{r}
write.table(VIP_allPathways, "metaboanalyst/significant-pathways-pairwise-VIP.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Identify pairwise VIP that overlap with VIP from significant pathways for each module

```{r}
module0_pairwiseVIPpathway <- rbind(VIP_module0 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module0 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module0 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module0 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module0") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module0_pairwiseVIPpathway
```

```{r}
module2_pairwiseVIPpathway <- rbind(VIP_module2 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module2 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module2 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module2 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module2") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module2_pairwiseVIPpathway
```

```{r}
module3_pairwiseVIPpathway <- rbind(VIP_module3 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module3 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module3 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module3 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module3") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module3_pairwiseVIPpathway
```
```{r}
module4_pairwiseVIPpathway <- rbind(VIP_module4 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module4 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module4 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module4 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module4") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module4_pairwiseVIPpathway
```

```{r}
module5_pairwiseVIPpathway <- rbind(VIP_module5 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module5 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module5 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module5 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module5") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module5_pairwiseVIPpathway
```

```{r}
module6_pairwiseVIPpathway <- rbind(VIP_module6 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module6 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module6 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module6 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module6") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module6_pairwiseVIPpathway
```

```{r}
module7_pairwiseVIPpathway <- rbind(VIP_module7 %>%
                                      filter(., metabolite %in% valineBiosynthesis) %>%
                                      mutate(., pathway = "valineBiosynthesis"),
                                    VIP_module7 %>%
                                      filter(., metabolite %in% alanineMetabolism) %>%
                                      mutate(., pathway = "alanineMetabolism"),
                                    VIP_module7 %>%
                                      filter(., metabolite %in% glutathioneMetabolism) %>%
                                      mutate(., pathway = "glutathioneMetabolism"),
                                    VIP_module7 %>%
                                      filter(., metabolite %in% arginineBiosynthesis) %>%
                                      mutate(., pathway = "arginineBiosynthesis")) %>%
  mutate(., module = "module7") #Identify metabolites present in pathways of interest. Combine information and add a column for module identity
module7_pairwiseVIPpathway
```

```{r}
pairwiseVIPpathway_allModules <- rbind(module0_pairwiseVIPpathway,
                                       module2_pairwiseVIPpathway,
                                       module3_pairwiseVIPpathway,
                                       module4_pairwiseVIPpathway,
                                       module5_pairwiseVIPpathway,
                                       module6_pairwiseVIPpathway,
                                       module7_pairwiseVIPpathway)
pairwiseVIPpathway_allModules
```

```{r}
write.table(pairwiseVIPpathway_allModules, "metaboanalyst/all-module-pairwise-VIP-pathways.txt", sep = "\t", quote = FALSE, row.names = FALSE)
```

#### Make heatmaps of each module's VIP

I'm only going to make heatmaps for the modules significantly correlated to a treatment x day condition.

##### Module 0

```{r}
#pdf("metaboanalyst/figures/module0_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module0$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module0_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module0$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module0_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module0$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 2

```{r}
#pdf("metaboanalyst/figures/module2_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module2$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module2_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module2$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module2_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module2$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 3

```{r}
#pdf("metaboanalyst/figures/module3_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module3$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module3_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module3$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module3_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module3$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 4

```{r}
#pdf("metaboanalyst/figures/module4_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module4$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module4_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module4$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module4_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module4$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 5

```{r}
#pdf("metaboanalyst/figures/module5_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module5$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module5_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module5$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module5_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module5$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 6

```{r}
#pdf("metaboanalyst/figures/module6_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module6$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module6_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module6$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module6_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module6$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

##### Module 7

```{r}
#pdf("metaboanalyst/figures/module7_VIP.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(., c(crab.ID, any_of(VIP_module7$metabolite))) %>%
            column_to_rownames(var = "crab.ID") %>%
            t(.) %>% as.data.frame(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, color = heatmapColors, annotation_col = metabolomicsAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module7_VIP-average.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module7$metabolite)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.) %>% log(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = TRUE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Abundance")
#dev.off()
```

```{r}
#pdf("metaboanalyst/figures/module7_VIP-zscore.pdf", height = 8.5, width = 11)
pheatmap((transMetabData %>%
            dplyr::select(treatment_day, any_of(VIP_module7$metabolite)) %>%
            group_by(., treatment_day) %>%
            mutate(across(where(is.numeric), log)) %>%
            mutate(across(where(is.numeric), scale)) %>%
            group_by(., treatment_day) %>%
            summarize_all(funs(mean(., na.rm = TRUE))) %>%
            column_to_rownames(var = "treatment_day") %>%
            t(.)), cluster_row = TRUE, show_rownames = TRUE, cluster_cols = FALSE, show_colnames = FALSE, scale = "row", color = heatmapColors, annotation_col = metabolomicsAvgAnnotation, annotation_colors = metabolomicsColors, name = "Z-Score")
#dev.off()
```

# ASCA

My WCNAs grouped compounds into module eigengenes and correlated those with temperature and physiological traits. I am going to use an ASCA to classify trends in metabolite abundance over temperature and time. Similar to the WCNA, this will provide me groups of compounds, but in a more statistically rigorous manner.

- [ALASCA tutorial](https://andjar.github.io/ALASCA/articles/ALASCA.html)
- [Ariana's lab notebook](https://ahuffmyer.github.io/ASH_Putnam_Lab_Notebook/Analysis-of-Variance-Simultaneous-Component-Analyses-ASCA-for-CEABIGR-project/)
- [Ariana's code from CEABiGR](https://github.com/sr320/ceabigr/blob/main/code/00-77-asca-exon.Rmd)

```{bash}
mkdir ASCA
mkdir ASCA/figures
```

## Format data

```{r}
metabolomics.ASCA <- rowwiseMetabData %>%
  log(.) %>%
  rownames_to_column(., var = "crab.ID") %>%
  pivot_longer(., cols = !crab.ID, names_to = "variable", values_to = "value") %>%
  left_join(metabolomicsMetadata, ., by = "crab.ID") %>%
  dplyr::select(-treatment_day) %>%
  mutate(., day = str_trim(day, side = "both")) %>%
  mutate(day = case_when(day == "3" ~ "04",
                         day != "3" ~ day)) %>%
  mutate(day = as.factor(day)) #Log-transform data. Pivot data longer for ASCA. Compound information should be "variable." Add metadata to dataframe with all compound information. Update metadata for day and reconvert day to a factor
head(metabolomics.ASCA) #Confirm changes
```

## Run model

```{r}
metabolomics.ASCAmodel <- ALASCA(df = metabolomics.ASCA,
                                 formula = value ~ day*treatment + (1|crab.ID),
                                 n_validation_runs = 100,
                                 validate = TRUE,
                                 reduce_dimensions = TRUE) #Run ASCA to identify day and temperature effects on compound abundance
```

## Interpret ASCA

```{r}
# pdf("ASCA/figures/scree-plot.pdf")
plot(metabolomics.ASCAmodel, component = 1:6, type = "scree") #Scree plot of variance explained by each PC. There were 6 components identified, with the sixth component explaining very little variance
# dev.off()
```

I'm now going to investigate the trends of the first five components, since those were the ones that explained any variance.

### Component 1

This component shows metabolites with consistently higher or lower abundance than 5ºC or 13ºC that change abundance (generally increasing) at 30ºC over time.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 1, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC1.pdf", height = 11, width = 8.5)
```
```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 1, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC1.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC1 <- get_loadings(metabolomics.ASCAmodel, component = 1, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC1)
```

```{r}
loadingsPC1.annot <- loadingsPC1[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC1.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC1.annot, "ASCA/loadings-PC1-annotations.csv", quote = FALSE, row.names = FALSE)
```

#### Metaboanalyst enrichment results

```{r}
topPathwayResultsPC1 <- read.csv("ASCA/PC1_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR)) %>%
  arrange(desc(enrichmentRatio)) %>%
  mutate(., order = rev(1:18)) #Import MetaboAnalyst results and modify for figure
head(topPathwayResultsPC1) #Confirm dataframe creation
```

```{r}
topPathwayResultsPC1 %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResultsPC1$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("ASCA/figures/PC1-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

### Component 2

This component highlights compounds in 5ºC that have similar abundance to 13ºC in day 4, but different in day 22.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 2, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 2, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC2.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC2 <- get_loadings(metabolomics.ASCAmodel, component = 2, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC2)
```

```{r}
loadingsPC2.annot <- loadingsPC2[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC2.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC2.annot, "ASCA/loadings-PC2-annotations.csv", quote = FALSE, row.names = FALSE)
```

### Component 3

Absolute abundance in 5ºC changes over time but 30ºC remains stable

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 3, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 3, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC3.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC3 <- get_loadings(metabolomics.ASCAmodel, component = 3, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC3)
```

```{r}
loadingsPC3.annot <- loadingsPC3[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC3.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC3.annot, "ASCA/loadings-PC3-annotations.csv", quote = FALSE, row.names = FALSE)
```

### Component 4

Absolute change in abundance over time in 30ºC is more extreme than at 5ºC.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 4, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC4.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 4, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC4.pdf", height = 11, width = 8.5)
```

```{r}
loadingsPC4 <- get_loadings(metabolomics.ASCAmodel, component = 4, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC4)
```

```{r}
loadingsPC4.annot <- loadingsPC4[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        filter(., p.adj < 0.05) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC4.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC4.annot, "ASCA/loadings-PC4-annotations.csv", quote = FALSE, row.names = FALSE)
```

#### Metaboanalyst enrichment results

```{r}
topPathwayResultsPC4 <- read.csv("ASCA/PC4_metaboanalyst-results/pathway_results.csv") %>%
  dplyr::rename(metabolite = "X") %>%
  mutate(., Observed = Hits/Total) %>%
  mutate(., Significant = case_when(FDR < 0.05 ~ "Yes",
                                    FDR >= 0.05 ~ "No")) %>%
  mutate(., enrichmentRatio = Hits/Expected) %>%
  mutate(., negLogFDR = -log(FDR)) %>%
  arrange(desc(enrichmentRatio)) %>%
  mutate(., order = rev(1:24)) #Import MetaboAnalyst results and modify for figure
head(topPathwayResultsPC4) #Confirm dataframe creation
```

```{r}
topPathwayResultsPC4 %>%
  ggplot(., aes(x = enrichmentRatio, y = as.factor(order), fill = Significant)) +
  geom_bar(stat = "identity") +
  labs(x = "Enrichment Ratio") +
  scale_y_discrete(name = "",
                   labels = rev(topPathwayResultsPC4$metabolite)) +
  scale_fill_manual(name = "FDR",
                    values = c("grey20", "springgreen4"),
                    labels = c(expression("">=0.05), expression(""<0.05))) +
  theme_classic(base_size = 15)
ggsave("ASCA/figures/PC4-enrichmentRatio.svg", heigh = 8.5, width = 11)
```

### Component 5

Absolute trends at 13ºC vary, absolute trends at 5ºC and 30ºC are similar.

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 5, type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-PC5.pdf", height = 11, width = 8.5)
```

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = 5, type = 'prediction', 
     n_limit = 10, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/prediction-plot-PC5.pdf", height = 11, width = 8.5)
```

Component 5 explains < 10% of the variance in this dataset. Additionally, the top 10 compounds strongly loaded onto PC5 have errors that cross 0. I think this makes the PC5 results difficult to interpret. I will go ahead and say that PCs that explain > 10% variance are my cutoff for analysis.

It's still good to have the list of loadings, so I'll save those anyways.

```{r}
loadingsPC5 <- get_loadings(metabolomics.ASCAmodel, component = 5, effect = 1, n_limit = 10) #Get loadings for the top 10 positively and negatively loaded compounds on the PC
head(loadingsPC5)
```

```{r}
loadingsPC5.annot <- loadingsPC5[[1]] %>%
  rename(metabolite = "covars") %>%
  mutate(metabolite = str_trim(metabolite, "both")) %>%
  dplyr::select(PC:metabolite) %>%
  left_join(., 
            y = rbind(t.test_30v5_day3 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day3"),
                      t.test_13v30_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v30_day22"),
                      t.test_13v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "13v5_day22"),
                      t.test_30v5_day22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "30v5_day22"),
                      t.test_5_day3v22 %>%
                        dplyr::select(metabolite, statistic, p.adj) %>%
                        mutate(metabolite = str_trim(metabolite, "both"),
                               comparison = "5_day3v22")) %>%
              arrange(metabolite), 
            by = "metabolite") %>%
  mutate(., respirationMetabolites = case_when(metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                               metabolite %in% respirationMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., calciumSignalingMetabolites = case_when(metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                    metabolite %in% calciumSignalingMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., coldProtectionMetabolites = case_when(metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == TRUE ~ "Y",
                                                  metabolite %in% coldProtectionMetabolites %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., valineBiosynthesis = case_when(metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                           metabolite %in% valineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., arginineBiosynthesis = case_when(metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == TRUE ~ "Y",
                                             metabolite %in% arginineBiosynthesis %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., glutathioneMetabolism = case_when(metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                              metabolite %in% glutathioneMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) %>%
  mutate(., alanineMetabolism = case_when(metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == TRUE ~ "Y",
                                          metabolite %in% alanineMetabolism %>% str_trim(., side = "both") == FALSE ~ "N")) #Obtain loadings dataframe from list and modify columns. Join with VIP information, a priori pathway information, and enrichment information.
head(loadingsPC5.annot) #Confirm changes
```

```{r}
write.csv(loadingsPC5.annot, "ASCA/loadings-PC5-annotations.csv", quote = FALSE, row.names = FALSE)
```

### Final figure

```{r}
plot(metabolomics.ASCAmodel, effect = 1, component = seq(1,4), type = 'effect', 
     n_limit = 5, 
     ribbon = TRUE,
     x_angle = 0, x_h_just= 0.5,
     flip_axis = TRUE,
     palette = rev(plotColors),
     my_theme = theme_classic(),
     x_label = "Day",
     group_label = "Temperature (ºC)")
ggsave("ASCA/figures/effect-plot-all-components.pdf", height = 11, width = 8.5)
```
